---
output: html_document
editor_options: 
  chunk_output_type: console
---

# Plotting diversity  
Some visualisations of alpha and beta diversity.

## Introduction
  
This chapter introduces you to a basic workflow for plotting alpha and beta diversity using phyloseq and ggplot. It provides examples for loading the phyloseq object from a previously saved .rds file, then how to calculate and include alpha diversity metrics into the sample_data of the phyloseq object, and finally how to plot some of these metrics, individually or combined.
  
Furthermore, beta diversity, i.e. four different type of ordinations are plotted. The examples, include non-metric dimensional scaling on Bray-Curtis dissimilaries, Principal Component Analysis (Aitchison distance), Redundancy Analysis ( Aitchison) and a Principal Coordinate Analysis on unweighted uniFrac distances.   
  
The stats behind the individual metrics are not discussed. If you want to learn more about the various options to analyse your amplicon-derived diversity, perhaps start here: https://sites.google.com/site/mb3gustame/home, the GUide to STatistical Analysis in Microbial Ecology (GUSTA ME). This is one of many other useful links and resources. Some others are listed in chapter \@ref(about). 
  
Prior to establishing alpha diversity indices, as well as Bray-Curtis dissimilarities, we will stick to rarefying sequences to a specific library size (library size = sum of sequence counts of a sample) to normalise the data and prevent bias due to library size. Usually, this is done using the smallest library size of all samples. If you want to learn more about rarefying I recommend this paper here: [Enhancing diversity analysis by repeatedly rarefying next generation sequencing data describing microbial communities](https://www.nature.com/articles/s41598-021-01636-1). We will follow the recommendations from the authors of this paper to resample without replacement.       
  
Other normalisation methods are also possible and it is up to you, the analyst, to understand and decide on the appropriate methods. I found centred-log ratios (Aitchison distances) also intuitive in interpretation, which are included into the workflow below in a PCA and RDA. Log-ratios are further discussed in the paper [Microbiome Datasets Are Compositional: And This Is Not Optional](https://doi.org/10.3389/fmicb.2017.02224).   
In fact, using different types of normalisation, dissimilarities or distances, while providing different types of information, may complement each other and help in interpretation of your data.    
  
### Prerequisites  
* a phyloseq object; either read in from a pre-saved `.rds` file or created as described in chapter \@ref(Qiime2R).   
   
Time to get started... 
  
## Workflow  
  
### Packages  
  
Install the following packages if not already.  
  
```{r packagesdiversity, message=FALSE, include=TRUE, echo=TRUE, warning=FALSE, message=FALSE}
# BiocManager::install("microbiome") If needed
library(phyloseq)
library(ggpubr)       # a handy helper package for ggplots
library(tidyverse)
theme_set(theme_bw())  # setting the theme for ggplots
library(vegan)
library(microbiome)

```

  \
  
### Load phyloseq object
For detail on how to create a phyloseq object please see chapter \@ref(Qiime2R).  
  
```{r loadps, include=TRUE, echo=TRUE, warning=FALSE, message=FALSE}
# reading in a previously saved phyloseq object
ps <- readRDS('ps_ProjectX_2022July')
ps  # get an overview of number of taxa and samples contained in the phyloseq object
```
  
  \
  
Check the metadata contained in the phyloseq object. This is the original metadata from the public repository where the FastQ files were downloaded from (Sequence Read Archives).  
  
There are 15 columns containing some character (chr), factor or integer (int) variables. 
  
```{r metadatacheck, include=TRUE, echo=TRUE, warning=FALSE, message=FALSE}
metadf <- data.frame(sample_data(ps))
str(metadf)
```
  
  \
  
### Pre-filter, rarefy and calculate diversity
  
Now the diversity indices can be calculated and added to the metadata of the phyloseq object for plotting. First, the abundances are pre-filtered (as shown in the previous chapter), then the ASV abundances are randomly resampled with `phyloseq::rarefy_even_depth`, such that all samples have the same number of ASVs.   
The diversity indices are then calculated with `phyloseq::estimate_richness` based on these evenly resampled abundances. The dataframe that is created containing the range of indices is then added to the existing metadata of the phyloseq for plotting.  
  
```{r alphadiversitypre, include=TRUE, echo=TRUE, warning=FALSE, message=FALSE, error=FALSE, cache=TRUE}
 
# some additional quality filtering - repeating steps from previous chapter. 
  
ps.flt = prune_taxa(taxa_sums(ps) >= 5, ps) #minimum reads per feature 
ps.flt  = subset_taxa(ps.flt , !is.na(Phylum) & !Phylum %in% c(""))
ps.flt <- ps.flt %>%
  subset_taxa(Kingdom == "Bacteria" & Family  != "Mitochondria" & Class   != "Chloroplast")
ps.flt 


# confirming minimum sample size
min(colSums(otu_table(ps.flt)))

# quantile summary of sample sizes
quantile(colSums(otu_table(ps.flt)))

# rarefy - required before measuring alpha diversity. 
# This step is to create an abundance table, where the all samples are randomly 
# resamples such that all samples have the same number of ASVs.  
# It removes any bias but number of re-samples is based on the sample with 
# smallest number of ASVs so one has to check and filter beforehand in case 
# there are samples with very different number of ASVs. I am generally going 
# by a 10x rule. If the lowest number of ASVs is 10x lower than the sample with
# highest number of ASVs, then I keep that sample in for alpha diversity assessments.  

ps_rare <- phyloseq::rarefy_even_depth(ps.flt, 
    sample.size = min(colSums(otu_table(ps.flt))),  
    rngseed = TRUE, 
    replace = FALSE, #without replacement
    trimOTUs = TRUE)

# check rarefied phyloseq object
# ps_rare (not run)

# phyloseq-class experiment-level object
# otu_table()   OTU Table:         [ 3801 taxa and 51 samples ]
# sample_data() Sample Data:       [ 51 samples by 15 sample variables ]
# tax_table()   Taxonomy Table:    [ 3801 taxa by 7 taxonomic ranks ]
 #phy_tree()    Phylogenetic Tree: [ 3801 tips and 3800 internal nodes ]

# Create diversity indices
div.df <- phyloseq::estimate_richness(ps_rare)
# add sample names
div.df$`#SampleID` <- phyloseq::sample_names(ps.flt)
# add diversity indices to phyloseq object
metadf <- data.frame(sample_data(ps.flt))  # export metadata first
metadf <- metadf %>% 
  rownames_to_column("#SampleID") %>%
  left_join(div.df, by = "#SampleID") %>% 
  column_to_rownames("#SampleID") # add the diversity indices (columns) to the metadata
# add pilou evenness (see calculation in "Numerical Ecology with R - Bocard et al")
metadf$pielou_ev <- div.df$Shannon/log(div.df$Observed)
# replace metadata with temp meta.df which includes the diversity indices
ps.flt@sam_data <- sample_data(metadf)  
ps_rare@sam_data <- sample_data(metadf)  
# this replaced the old metadata with new metadata that includes the diversity 
# indices for this phyloseq object 
str(data.frame(sample_data(ps.flt)))
```


### Colors  
  
```{r cols}
# You can check Hex codes with help of https://htmlcolorcodes.com/

cols <- c("#F9E79F", "#FDEBD0", "#F6DDCC",  "#E5E8E8", "#CCD1D1", "#AAB7B8", 
          "#707B7C", "#566573","#82E0AA", "#641E16", "#8E44AD", "#E74C3C")

# you can also name each colour according to your treatment. 
# This will help ggplot to associated the same colour for each treatment across
# different ggplots. For example, if we want to show different colour by location 
# in this dataset: 
names(cols)  <- unique((data.frame(sample_data(ps.flt)))$Location)
cols
```

### Richness
```{r richness,  message=FALSE, warning=FALSE, echo=T}

p1 <- data.frame(phyloseq::sample_data(ps.flt)) %>%
  ggpubr::ggboxplot(., x = "Location", 
                    y = "Observed", 
                    fill = "Location", 
                    ylab = "Richness", 
                    xlab = "Location", 
                    palette = cols) +
    theme_bw() +
    theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) +
    geom_jitter(width = 0.1) +
   theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
#    theme(axis.title.x=element_blank(),  # Options
#        axis.text.x=element_blank(),
#        axis.ticks.x=element_blank()) +
  ggpubr::stat_compare_means(method = "kruskal",label.x = 2)
p1

```

### Shannon
```{r shannon,  message=FALSE, warning=FALSE, echo=T}
# Shannon
p2 <- data.frame(phyloseq::sample_data(ps.flt)) %>%
  ggpubr::ggboxplot(., x = "Location", 
                    y = "Shannon", 
                    fill = "Location", 
                    ylab = "Shannon diversity (H)", 
                    xlab = "Location", 
                    palette = cols) +
    theme_bw() +
    theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) +
    geom_jitter(width = 0.1) +
   theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
#    theme(axis.title.x=element_blank(),  # Options
#        axis.text.x=element_blank(),
#        axis.ticks.x=element_blank()) +
  ggpubr::stat_compare_means(method = "kruskal",label.x = 2)
p2
```

### Pielou's evenness
```{r pielou,  message=FALSE, warning=FALSE, echo=T }
# Pielou
p3 <- data.frame(phyloseq::sample_data(ps.flt)) %>%
  ggpubr::ggboxplot(., x = "Location", 
                    y = "pielou_ev", 
                    fill = "Location", 
                    ylab = "Pielou Evenness (J)", 
                    xlab = "Location", 
                    palette = cols) +
    theme_bw() +
    theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) +
    geom_jitter(width = 0.1) +
   theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
#    theme(axis.title.x=element_blank(),  # Options
#        axis.text.x=element_blank(),
#        axis.ticks.x=element_blank()) +
  ggpubr::stat_compare_means(method = "kruskal",label.x = 2)
p3
```

### Combined plotting 
```{r plot, echo=T, fig.asp=0.45, fig.width=10, message=FALSE, warning=FALSE, include=TRUE}
#pdf(NULL)
g1 <- ggpubr::ggarrange(p1, p2, p3, common.legend = TRUE, nrow = 1, legend = 'top')
#x = dev.off()
g1
```
  
  \
  
### NMDS  
```{r, nmds, echo=TRUE, message=FALSE, warning=FALSE, results=FALSE, cache=FALSE,fig.asp=0.8, fig.width=6}
physeqNMDS <-  ps_rare  # We are using the rarefied abundances as explained in the intro 

#ordination
ordination <- phyloseq::ordinate(physeqNMDS  , "NMDS", distance = 'bray') # using Bray-Curtis dissimilarities here. 

label <- round(ordination$stress,3) # stress label
#plotting ordination
NMDSplot <- phyloseq::plot_ordination(physeqNMDS, 
                                      ordination , 
                                      color = "Location", 
                                      #shape = "Reactor",    
                                      # option to add shapes as data symbols
                                      title = "NMDS") +
     geom_point(aes(size = pielou_ev), alpha = 0.75) +   # shape size in proportion to Pielou Evenness, also make transparent 
     annotate(geom = 'text', label = paste("Bray-Curtis, Stress = ", label), 
             x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3)  +
     scale_color_manual(values = cols)

NMDSplot
```


### PCA  
```{r PCA,  message=FALSE, warning=FALSE, echo=T, include=TRUE,fig.asp=0.8, fig.width=6}

physeqPCA <-  ps.flt  # just to keep it separate.

# normalisation and transform using the microbiome package
# I chose a centred log-ratio transform and a principal component analysis in a euclidean space.
physeqPCA <- microbiome::transform(physeqPCA, "clr")

#ordination
ordination <- phyloseq::ordinate(physeqPCA , "RDA") # useing the centred-log transformed abundances

#plotting ordination
PCAplot <- phyloseq::plot_ordination(physeqPCA, 
                                      ordination , 
                                      color = "Location", 
                                      #shape = "Reactor", 
                                     # option to add shapes as data symbols
                                      title = "Principal Component Analysis") +
     geom_point(aes(size = pielou_ev)) +    # # shape size in proportion to Pielou Evenness
    annotate(geom = 'text', label = paste("Aitchison distances"), 
             x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3)  + 
    scale_color_manual(values = cols)

PCAplot

```
  
  \
   
### RDA  
```{r rda,  message=FALSE, warning=FALSE, echo=T, include=TRUE,fig.asp=0.8, fig.width=6 }
# PERCENT FILTERVALUE FOR ALL RDAs
pcnt <- 0.25
# FORMULA FOR ALL RDAs
form <- formula(~ Location)
# labels for arrows
# pos for arrows
xarw <- 5
yarw <- 5
# arrow length map
larw <- 1.5
# arrow length map for chromosol rda
larwch <- 1
# fixed width of x axis
xaxis = c(-6.75, 6.5)
# position of annotation
textposx = c(-4.4)
# arrow color
arrowcolor <- "grey13"


physeqPCA <- ps.flt # excluding Heat, H2O2, reference and blank

# normalisation and transform
physeqPCA <- microbiome::transform(physeqPCA, "clr")

#ordination
ordination <- phyloseq::ordinate(physeqPCA, "RDA", "bray", formula = form )

RDAplot <- phyloseq::plot_ordination(physeqPCA, ordination , 
                                color = "Location")  +
  geom_hline(yintercept=0, linetype="dashed", color = "gray") +
  geom_vline(xintercept=0, linetype="dashed", color = "gray") +    # shape size reflective of Pielou  
     annotate(geom = 'text', label = paste("Aitchison distances"), 
             x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3) +
  geom_point(aes(size = pielou_ev)) +        # shape size in proportion to Pielou Evenness
  #scale_colour_gradient(low = colourlow, high = colourhigh) +
  ggtitle("RDA Abundances (clr) ~ Location") +
  scale_color_manual(values = cols)


# Now add the environmental variables as arrows to either of these p1 or p2
arrowmat <- vegan::scores(ordination, display = "bp")
arrowmat <- data.frame(arrowmat)
#rownames(arrowmat) <- arrowlabel
# Add labels, make a data.frame
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrowdf$labels <- arrowdf$labels %>% gsub("Location", "", .) # remove the string "Location" from the labels. 
# Define the arrow aesthetic mapping
arrow_map <- aes(xend = larw * RDA1,
                 yend = larw * RDA2,
                 x = 0,
                 y = 0,
                 shape = NULL,
                 color = NULL,
                 label = labels)

label_map <- aes(x = xarw * RDA1,
                 y = yarw * RDA2,
                 shape = NULL,
                 color = NULL,
                 label = labels)

arrowhead = arrow(length = unit(0.02, "npc"))

RDAplot <-RDAplot +
  geom_segment(
    mapping = arrow_map,
    size = .5,
    data = arrowdf,
    color = arrowcolor,
    arrow = arrowhead)  +
  geom_text(
   mapping = label_map,
  size = 4,
  data = arrowdf,
  show.legend = FALSE)


RDAplot

# save figure for publication
# ggsave("RDA.png", height=6, width=7.5, units='in', dpi=600)
#knitr::include_graphics("./RDA.png")


```
  
  \
  
### Unweighted uniFrac  
UniFrac distances are derived from phylogenetic distances between ASVs, which means that a phylogenetic tree needs to be included into the phyloseq object.  
  
```{r, unifrac, echo=TRUE, message=FALSE, warning=FALSE, cache=FALSE,fig.asp=0.8, fig.width=6}
physeqPcOA <-  ps.flt  # just to keep it separate.

#ordination
ordination <- phyloseq::ordinate(physeqPcOA  , 
                                 "PCoA", 
                                 distance = 'unifrac') # using just presence and absence (unweighted) dissimilarities here. 

#plotting ordination
PCOAplot <- phyloseq::plot_ordination(physeqPcOA, 
                                      ordination , 
                                      color = "Location", 
                                      #shape = "Reactor",    
                                      # option to add shapes to data symbols
                                      title = "Principal Coordinate Analysis") +
     geom_point(aes(size = pielou_ev)) +    # shape size reflective of Pielou  
     annotate(geom = 'text', label = paste("unweighted UniFrac"), 
             x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3)  +
     scale_color_manual(values = cols)

PCOAplot
```

### Combined plotting 
```{r plotbeta, echo=T, fig.asp=1, fig.width=10, message=FALSE, warning=FALSE, include=TRUE}
#pdf(NULL)
g2 <- ggpubr::ggarrange(NMDSplot, PCAplot, RDAplot, PCOAplot,
                        common.legend = TRUE, nrow = 2, ncol = 2, legend = 'top')
#x = dev.off()
g2
```
  
  \
  
### Save plots to png and pdf
```{r}
# either as png or pdf
ggsave("alphadiversity.png", plot = (g1), height=4, width=10, units='in', dpi=600)
ggsave("alphadiversity.pdf", plot = (g1), height=4, width=10, units='in', dpi=600)

ggsave("betadiversity.png", plot = (g2), height=10, width=10, units='in', dpi=600)
ggsave("betadiversity.pdf", plot = (g2), height=10, width=10, units='in', dpi=600)
```
  
  
**Note:**
This code is an amalgamation from various sources. Apart from putting it together into a pipeline I do not take credit for it.
