[*An*]{style="color:#C00000"} [\_\_ \_\_]{style="color:#C00000"} [*introduction to alpha and beta diversity analysis and the pitfalls you need to know*]{style="color:#C00000"}

[Dr Jen WOOD \|]{style="color:#C00000"} [*Microbial community ecologist*]{style="color:#C00000"}

[La Trobe University]{style="color:#C00000"}

![](img/Diversity_workshop_JWforwebsite0.jpg)

![](img/Diversity_workshop_JWforwebsite1.png)

[\_\_ \_\_]{style="color:#C00000"} [JW_ilikedirt]{style="color:#C00000"}

[Jen.Wood]{style="color:#C00000"} @latrobe [.edu.au]{style="color:#C00000"}

[**W**]{style="color:#C00000"} [[https://jenwoodmicro\\.wordpress\\.com](https://jenwoodmicro.wordpress.com){.uri}]{style="color:#C00000"} [/]{style="color:#C00000"}

![](img/Diversity_workshop_JWforwebsite2.png)

![](img/Diversity_workshop_JWforwebsite3.png)

# Microbial Diversity, how do you describe it?

How many species $ESVs$ are there?

Which species are common and which are rare?

How many species do *two* communities share?

How many species do *two* communities have different?

Do *two* communities share the same common and rare species?

**Definition:** Alpha diversity is the diversity of each site $local species pool$.

**Common Metrics** : Shannon-Wiener, Simpson diversity

**Affected by:** Species **RICHNESS** and community **EVENNESS**

**RICHNESS** : number of species in a community

*Not* affected by the number of INDIVIDUALS $READS/ABUNDACE$.

*Not* affected by taxonomy

The more species rich a community is the more diverse that community is

**EVENNESS** : how evenly individuals are distributed among the species in a community *IS* affected by the number of INDIVIDUALS $READS/ABUNDACE$.

*Not* affected by taxonomyThe more even a community is the more diverse that community is

[**Community richness:**]{style="color:#E52212"}

The more species-rich a community is the more diverse that community is

[*Not*]{style="color:#E52212"} [\_ affected by total individuals \_]{style="color:#E52212"} Two lakes with identical **species richness** of 5

Just one is more crowded than the other

![](img/Diversity_workshop_JWforwebsite4.png)

![](img/Diversity_workshop_JWforwebsite5.png)

![](img/Diversity_workshop_JWforwebsite6.png)

![](img/Diversity_workshop_JWforwebsite7.png)

![](img/Diversity_workshop_JWforwebsite8.png)

![](img/Diversity_workshop_JWforwebsite9.png)

![](img/Diversity_workshop_JWforwebsite10.png)

![](img/Diversity_workshop_JWforwebsite11.png)

![](img/Diversity_workshop_JWforwebsite12.png)

![](img/Diversity_workshop_JWforwebsite13.png)

![](img/Diversity_workshop_JWforwebsite14.png)

![](img/Diversity_workshop_JWforwebsite15.png)

![](img/Diversity_workshop_JWforwebsite16.png)

![](img/Diversity_workshop_JWforwebsite17.png)

![](img/Diversity_workshop_JWforwebsite18.png)

![](img/Diversity_workshop_JWforwebsite19.png)

![](img/Diversity_workshop_JWforwebsite20.png)

![](img/Diversity_workshop_JWforwebsite21.png)

![](img/Diversity_workshop_JWforwebsite22.png)

![](img/Diversity_workshop_JWforwebsite23.png)

![](img/Diversity_workshop_JWforwebsite24.png)

![](img/Diversity_workshop_JWforwebsite25.png)

![](img/Diversity_workshop_JWforwebsite26.png)

![](img/Diversity_workshop_JWforwebsite27.png)

![](img/Diversity_workshop_JWforwebsite28.png)

![](img/Diversity_workshop_JWforwebsite29.png)

![](img/Diversity_workshop_JWforwebsite30.png)

![](img/Diversity_workshop_JWforwebsite31.png)

![](img/Diversity_workshop_JWforwebsite32.png)

![](img/Diversity_workshop_JWforwebsite33.png)

![](img/Diversity_workshop_JWforwebsite34.png)

![](img/Diversity_workshop_JWforwebsite35.png)

![](img/Diversity_workshop_JWforwebsite36.png)

![](img/Diversity_workshop_JWforwebsite37.png)

![](img/Diversity_workshop_JWforwebsite38.png)

![](img/Diversity_workshop_JWforwebsite39.png)

![](img/Diversity_workshop_JWforwebsite40.png)

![](img/Diversity_workshop_JWforwebsite41.png)

![](img/Diversity_workshop_JWforwebsite42.png)

![](img/Diversity_workshop_JWforwebsite43.png)

![](img/Diversity_workshop_JWforwebsite44.png)

![](img/Diversity_workshop_JWforwebsite45.png)

![](img/Diversity_workshop_JWforwebsite46.png)

![](img/Diversity_workshop_JWforwebsite47.png)

![](img/Diversity_workshop_JWforwebsite48.png)

![](img/Diversity_workshop_JWforwebsite49.png)

![](img/Diversity_workshop_JWforwebsite50.png)

![](img/Diversity_workshop_JWforwebsite51.png)

![](img/Diversity_workshop_JWforwebsite52.png)

![](img/Diversity_workshop_JWforwebsite53.png)

![](img/Diversity_workshop_JWforwebsite54.png)

![](img/Diversity_workshop_JWforwebsite55.png)

![](img/Diversity_workshop_JWforwebsite56.png)

![](img/Diversity_workshop_JWforwebsite57.png)

![](img/Diversity_workshop_JWforwebsite58.png)

![](img/Diversity_workshop_JWforwebsite59.png)

[*Not*]{style="color:#E52212"} [\_ affected by taxonomy\_]{style="color:#E52212"}

Two communities with an identical **OTU richness** of 4 but with NO species in common

![](img/Diversity_workshop_JWforwebsite60.png)

![](img/Diversity_workshop_JWforwebsite61.png)

![](img/Diversity_workshop_JWforwebsite62.png)

![](img/Diversity_workshop_JWforwebsite63.png)

![](img/Diversity_workshop_JWforwebsite64.png)

![](img/Diversity_workshop_JWforwebsite65.png)

![](img/Diversity_workshop_JWforwebsite66.png)

![](img/Diversity_workshop_JWforwebsite67.png)

![](img/Diversity_workshop_JWforwebsite68.png)

![](img/Diversity_workshop_JWforwebsite69.png)

FORREST

Richness = 4

SAVANNA

Richness = 4

[**Community evenness**]{style="color:#E52212"}

Describes how evenly the number of individuals are distributed among the species in a given community

Completely even community

Uneven community

Uneven community with over dominant species

![](img/Diversity_workshop_JWforwebsite70.png)

![](img/Diversity_workshop_JWforwebsite71.png)

![](img/Diversity_workshop_JWforwebsite72.png)

Abundance of individuals

Abundance of individuals

Abundance of individuals

![](img/Diversity_workshop_JWforwebsite73.png)

![](img/Diversity_workshop_JWforwebsite74.png)

![](img/Diversity_workshop_JWforwebsite75.png)

![](img/Diversity_workshop_JWforwebsite76.png)

![](img/Diversity_workshop_JWforwebsite77.png)

![](img/Diversity_workshop_JWforwebsite78.png)

![](img/Diversity_workshop_JWforwebsite79.png)

![](img/Diversity_workshop_JWforwebsite80.png)

![](img/Diversity_workshop_JWforwebsite81.png)

![](img/Diversity_workshop_JWforwebsite82.png)

![](img/Diversity_workshop_JWforwebsite83.png)

![](img/Diversity_workshop_JWforwebsite84.png)

![](img/Diversity_workshop_JWforwebsite85.png)

![](img/Diversity_workshop_JWforwebsite86.png)

![](img/Diversity_workshop_JWforwebsite87.png)

![](img/Diversity_workshop_JWforwebsite88.png)

![](img/Diversity_workshop_JWforwebsite89.png)

![](img/Diversity_workshop_JWforwebsite90.png)

![](img/Diversity_workshop_JWforwebsite91.png)

![](img/Diversity_workshop_JWforwebsite92.png)

Species $1\,2… n$

Species $1\,2… n$

Species $1\,2… n$

**For this pair of communities: which is more diverse and why**

![](img/Diversity_workshop_JWforwebsite93.png)

![](img/Diversity_workshop_JWforwebsite94.png)

![](img/Diversity_workshop_JWforwebsite95.png)

![](img/Diversity_workshop_JWforwebsite96.png)

[**Shannon-Weiner Diversity**]{style="color:#E52212"}

The **Richness** and **Evenness** of a community contribute to its **Diversity**

This is a score that can be calculated a number of ways.

The Shannon-Wiener equation below is one of the most common methods.

**In general MORE richness = MORE diversity**

**And MORE evenness = MORE diversity**

 = Sum of

*p* i = Percentage in decimal form

Ln$x$ = natural logarithm. Use a calculator!

\* = multiply

**In general MORE richness = MORE diversity**

**And MORE evenness = MORE diversity**

|      Species       | Number of individuals (p) |  pi  | Ln(pi) | pi\*Ln(pi) | -pi\*Ln(pi) |
|:------------------:|:-------------------------:|:----:|:------:|:----------:|:-----------:|
|   Geobacter spp.   |            23             | 0.37 | -0.99  |   -0.37    |    0.37     |
| Desulfovibrio spp. |            27             | 0.44 | -0.83  |   -0.36    |    0.36     |
|  Clostridium spp.  |            12             | 0.19 | -1.64  |   -0.32    |    0.32     |
|        Sum:        |            62             |      |        |            |    1.05     |

 = Sum of

*p* i = Percentage in decimal form

Ln$x$ = natural logarithm. Use a calculator!

\* = multiply

**In general MORE richness = MORE diversity**

[**And MORE evenness = MORE diversity**]{style="color:#E52212"}

|     Speceis(i)     | Number of individuals (p) |  pi  | Ln(pi) | pi\*Ln(pi) | -pi\*Ln(pi) |
|:------------------:|:-------------------------:|:----:|:------:|:----------:|:-----------:|
|   Geobacter spp.   |            52             | 0.84 | -0.18  |   -0.15    |    0.15     |
| Desulfovibrio spp. |             7             | 0.11 | -2.18  |   -0.25    |    0.25     |
|  Clostridium spp.  |             2             | 0.03 | -3.43  |   -0.11    |    0.11     |
|        Sum:        |            62             |      |        |            |    0.51     |

**Definition** : The variation in species composition among sites

**Metrics** : $dis$similarity indices such as [Bray-Curtis $BC$]{style="color:#C00000"} , Unifrac, Aitchison

**Affected by** : number of ESVs in common, number of ESVs not in common, similarity of shared ESV **abundances** between plots\*

Where:

i & j are the two sites,

Si is the total number of specimens counted on site i,

Sj is the total number of specimens counted on site j,

Cij is the sum of only the lesser counts for each species found in both sites.

\*When data is abundance weighted $i\.e\. not\- binary transformed$

![](img/Diversity_workshop_JWforwebsite97.jpg)

**Two communities with identical richness,** **diveristy** \_\_ and biomass can \_\_ **still be a bit dissimilar**

Tank 1Richness = 5 \| Net abundance = 10

Tank 2Richness = 5 \| Net abundance = 10

![](img/Diversity_workshop_JWforwebsite98.png)

![](img/Diversity_workshop_JWforwebsite99.png)

![](img/Diversity_workshop_JWforwebsite100.png)

![](img/Diversity_workshop_JWforwebsite101.png)

![](img/Diversity_workshop_JWforwebsite102.png)

![](img/Diversity_workshop_JWforwebsite103.png)

![](img/Diversity_workshop_JWforwebsite104.png)

![](img/Diversity_workshop_JWforwebsite105.png)

![](img/Diversity_workshop_JWforwebsite106.png)

![](img/Diversity_workshop_JWforwebsite107.png)

![](img/Diversity_workshop_JWforwebsite108.png)

![](img/Diversity_workshop_JWforwebsite109.png)

![](img/Diversity_workshop_JWforwebsite110.png)

![](img/Diversity_workshop_JWforwebsite111.png)

![](img/Diversity_workshop_JWforwebsite112.png)

![](img/Diversity_workshop_JWforwebsite113.png)

![](img/Diversity_workshop_JWforwebsite114.png)

![](img/Diversity_workshop_JWforwebsite115.png)

![](img/Diversity_workshop_JWforwebsite116.png)

![](img/Diversity_workshop_JWforwebsite117.png)

![](img/Diversity_workshop_JWforwebsite118.png)

![](img/Diversity_workshop_JWforwebsite119.png)

![](img/Diversity_workshop_JWforwebsite120.png)

![](img/Diversity_workshop_JWforwebsite121.png)

![](img/Diversity_workshop_JWforwebsite122.png)

2$2 \+ 2 \+ 2 \+ 2 \+ 1$ = 18 = 0.9

10 + 10 20

**Two communities with identical richness, diversity and biomass** **can be highly dissimilar**

Tank 1Richness = 3 \| Net abundance = 5

Tank 2Richness = 3 \| Net abundance = 5

![](img/Diversity_workshop_JWforwebsite123.png)

![](img/Diversity_workshop_JWforwebsite124.png)

![](img/Diversity_workshop_JWforwebsite125.png)

![](img/Diversity_workshop_JWforwebsite126.png)

![](img/Diversity_workshop_JWforwebsite127.png)

![](img/Diversity_workshop_JWforwebsite128.png)

![](img/Diversity_workshop_JWforwebsite129.png)

![](img/Diversity_workshop_JWforwebsite130.png)

![](img/Diversity_workshop_JWforwebsite131.png)

![](img/Diversity_workshop_JWforwebsite132.png)

![](img/Diversity_workshop_JWforwebsite133.png)

![](img/Diversity_workshop_JWforwebsite134.png)

![](img/Diversity_workshop_JWforwebsite135.png)

![](img/Diversity_workshop_JWforwebsite136.png)

![](img/Diversity_workshop_JWforwebsite137.png)

2$0 \+ 0 \+ 0 \+ 0 \+ 1$ = 2 = 0.2

5 + 5 10

[**Changes in beta diversity may not track with changes in alpha diversity**]{style="color:#E52212"}

The variation in species composition among sites can arise from:

Species replacement $i\.e\. community turnover$. Due to the tendency of species to replace each other over an ecological gradient

Species differences $i\.e\. community nestedness$. Due to one community having more species present than another

[**Species replacement**]{style="color:#E52212"}

Tank 1Richness = 2 \| Net abundance = 5

Tank 2Richness = 2 \| Net abundance = 5

![](img/Diversity_workshop_JWforwebsite138.png)

![](img/Diversity_workshop_JWforwebsite139.png)

![](img/Diversity_workshop_JWforwebsite140.png)

![](img/Diversity_workshop_JWforwebsite141.png)

![](img/Diversity_workshop_JWforwebsite142.png)

![](img/Diversity_workshop_JWforwebsite143.png)

![](img/Diversity_workshop_JWforwebsite144.png)

![](img/Diversity_workshop_JWforwebsite145.png)

![](img/Diversity_workshop_JWforwebsite146.png)

![](img/Diversity_workshop_JWforwebsite147.png)

![](img/Diversity_workshop_JWforwebsite148.png)

![](img/Diversity_workshop_JWforwebsite149.png)

![](img/Diversity_workshop_JWforwebsite150.png)

![](img/Diversity_workshop_JWforwebsite151.png)

![](img/Diversity_workshop_JWforwebsite152.png)

2$0 \+ 0 \+ 0 \+ 0 \+ 1$ = 2 = 0.2

5 + 5 10

[**Species**]{style="color:#E52212"} [**nestedness**]{style="color:#E52212"}

Tank 1Richness = 5 \| Net abundance = 10

Tank 2Richness = 2 \| Net abundance = 5

![](img/Diversity_workshop_JWforwebsite153.png)

![](img/Diversity_workshop_JWforwebsite154.png)

![](img/Diversity_workshop_JWforwebsite155.png)

![](img/Diversity_workshop_JWforwebsite156.png)

![](img/Diversity_workshop_JWforwebsite157.png)

![](img/Diversity_workshop_JWforwebsite158.png)

![](img/Diversity_workshop_JWforwebsite159.png)

![](img/Diversity_workshop_JWforwebsite160.png)

![](img/Diversity_workshop_JWforwebsite161.png)

![](img/Diversity_workshop_JWforwebsite162.png)

![](img/Diversity_workshop_JWforwebsite163.png)

![](img/Diversity_workshop_JWforwebsite164.png)

![](img/Diversity_workshop_JWforwebsite165.png)

![](img/Diversity_workshop_JWforwebsite166.png)

![](img/Diversity_workshop_JWforwebsite167.png)

![](img/Diversity_workshop_JWforwebsite168.png)

![](img/Diversity_workshop_JWforwebsite169.png)

![](img/Diversity_workshop_JWforwebsite170.png)

![](img/Diversity_workshop_JWforwebsite171.png)

![](img/Diversity_workshop_JWforwebsite172.png)

2$0 \+ 2 \+ 2 \+ 0 \+ 0$ = 8 = 0.5

10 + 5 15

[**Beta Diversity analysis workflow**]{style="color:#E52212"}

Data transformations?

Raw data matrix $ESV table$

$Dis$similarity matrix

![](img/Diversity_workshop_JWforwebsite173.png)

*Do want to do an unweighted ordination? Presence-absence transform*

|       | S1  | S2  | S3  |
|:-----:|:---:|:---:|:---:|
| ESV 1 | 50  | 60  | 10  |
| ESV 2 |  5  | 10  |  2  |
| ESV 3 |  3  |  7  |  3  |
| ESV 4 |  0  |  0  |  1  |
| ESV 5 |  0  |  0  |  2  |

![](img/Diversity_workshop_JWforwebsite174.wmf)

![](img/Diversity_workshop_JWforwebsite175.jpg)

ORDINATION IS NOT AN END POINT. IT DOES NOT TEST A HYPOTHESIS IT SHOULD BE USED TO GENERATE A HYPOTHESIS. IT IS A START POINT

Statistical testsANOSIM/PERMANOVER

[**Beta Diversity graphics**]{style="color:#C00000"}

One table ordinations:

$NMDS\, PCoA$

Two table ordinations:

$CCA\, RDA$

![](img/Diversity_workshop_JWforwebsite176.wmf)

![](img/Diversity_workshop_JWforwebsite177.jpg)

![](img/Diversity_workshop_JWforwebsite178.png)

[**Beta Diversity graphics: NMDS**]{style="color:#C00000"}

Exactly the same data in an NMDS and in a PCoA using Bray-Curtis metric: what's the difference?

![](img/Diversity_workshop_JWforwebsite179.png)

![](img/Diversity_workshop_JWforwebsite180.png)

[**Beta Diversity graphics: ordinations**]{style="color:#C00000"}

**NMDS:** non-Metric Multidimensional Scaling

**What is happening:**

1\. A half matrix containing the dissimilarity scores between each pair of samples in calculated.

2\. The dissimilarity relationship between all the samples can be explained in n-dimensional space. Where n = number of samples -1

3\. The computer iteratively tries to project this multidimensional relationship space into 2D. If it can do this and *preserve the relationships* , stress \< 0.2. If it can't do this stress \> 0.2 and the ordination is basically random collection of dots

Relationship between 2 samples completely described by 1 dimension

Relationship between 3 samples completely described by 2 **orthogonal** dimensions

![](img/Diversity_workshop_JWforwebsite181.png)

Relationship between n samples completely described by **n-1** **orthogonal** dimensions.But ordination is attempting to represent this in 2D

[**Beta Diversity graphics: NMDS**]{style="color:#C00000"}

![](img/Diversity_workshop_JWforwebsite182.png)

**NMDS:** non-Metric Multidimensional Scaling

**Interpretation:** the closer two points are in 2D space the more similar they are. i.e. we looking at grouping

**Axis:** are arbitrary

**Valid if:** Kruskal stress values \< 0.2

**Pros:** lets the data "speak for itself". Way the ordination is drawn is completely analogous to the way ANOSIM/PERMANOVA stats are generated.

**Cons:** ordinations are not always valid

[**Beta Diversity graphics:**]{style="color:#C00000"} [**PCoA**]{style="color:#C00000"} [\_\_ \_\_]{style="color:#C00000"}

![](img/Diversity_workshop_JWforwebsite183.jpg)

**PCoA** **:** Principal Coordinates analysis

**What is happening:**

1\. A half matrix containing the dissimilarity scores between each pair of samples in calculated and 'placed' in n-dimensional space

2\. An axis is draw through the data in the dimension that has the largest variation

3\. Data is projected onto this axis

4\. The process is repeated for the direction through the data that has the second greatest amount of variation so the data are now projected onto these two new axis, the PCoA axis

![](img/Diversity_workshop_JWforwebsite184.jpg)

A 2D cloud of data where distances = similarity. Axis through direction of greatest variation $greatest data spread$ and projection onto that axis. Some relationship info is lost.

![](img/Diversity_workshop_JWforwebsite185.jpg)

Axis through the two greatest directions of variation in data cloud

![](img/Diversity_workshop_JWforwebsite186.jpg)

Projection of data onto the new axis. NB no info is lost in example b/c we in 2D

[**Beta Diversity graphics:**]{style="color:#C00000"} [**PCoA**]{style="color:#C00000"} [\_\_ \_\_]{style="color:#C00000"}

![](img/Diversity_workshop_JWforwebsite187.png)

![](img/Diversity_workshop_JWforwebsite188.png)

**PCoA** **:** Principal Coordinates analysis

**Interpretation:** order of data along axis correspond to largest variation in data set

**Axis:** percentage of variation in the 'data cloud'

**Pros:** always valid - no stress values

May correlate treatment effect to a tangible level of variation in the data

**Cons:** Axis don't always explain lots of variationCommunities that have ESV values that introduce large amounts of variation strongly influence the ordination, making patterns of other communities less visible. May need to remove these communities

[**Beta Diversity graphics: Dendrogram**]{style="color:#C00000"}

![](img/Diversity_workshop_JWforwebsite189.wmf)

**Dendrogram:** **Interpretation:** branch points indicate how similar the samples below are. NB the top-bottom order is often arbitrary EVERY BRACH POINT CAN BE ROTATED

**Axis:** amount of dissimilarity between groups at branch point

**Pros:** you'll never have data points on top of each other $as can happen with ordinations when there are outliers or really high variation b/w groups$

**Cons:** not always easy to see patterns

[**Testing Beta Diversity: PERMANOVA**]{style="color:#E52212"}

A non-parametric alternative to MANOVA $also called NPMANOVA$

Generates a pseudo-F stat and F table with p values $like an ANOVA$

Test hypothesis that there is no difference in group centriods

![](img/Diversity_workshop_JWforwebsite190.png)

[**Testing Beta Diversity: PERMANOVA**]{style="color:#E52212"}

PERMANOVA is not done on the output of an ordination but rather on the underlying distance matrices

![](img/Diversity_workshop_JWforwebsite191.png)

|     | R1  | R2  | R3  | R4  | R5  | S1  | S2  | S3  | S4  | S5  |
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
| R1  | \-  |     |     |     |     |     |     |     |     |     |
| R2  | 0.6 | \-  |     |     |     |     |     |     |     |     |
| R3  | 0.7 | 0.8 | \-  |     |     |     |     |     |     |     |
| R4  | 0.9 | 0.8 | 0.9 | \-  |     |     |     |     |     |     |
| R5  | 0.9 | 0.6 | 0.7 | 0.5 | \-  |     |     |     |     |     |
| S1  | 0.2 | 0.4 | 0.3 | 0.4 | 0.2 | \-  |     |     |     |     |
| S2  | 0.1 | 0.3 | 0.5 | 0.4 | 0.3 | 0.7 | \-  |     |     |     |
| S3  | 0.3 | 0.4 | 0.3 | 0.3 | 0.3 | 0.9 | 0.7 | \-  |     |     |
| S4  | 0.2 | 0.5 | 0.1 | 0.4 | 0.4 | 0.8 | 0.6 | 0.9 | \-  |     |
| S5  | 0.1 | 0.2 | 0.1 | 0.2 | 0.5 | 0.9 | 0.9 | 0.7 | 0.6 | \-  |

Where: *SS* *B* = sum of squared differences between dissimilarity scores between groups. Where: [\_\_ \_\_]{style="color:#93CDDD"} *SS* *w* [\_\_ \_\_]{style="color:#93CDDD"} = sum of squared differences between dissimilarity scores within groups. $a\-1$ and $N\-a$ are terms relating to df associated with treatments and with residuals respectively

[**Testing Beta Diversity: Beta-**]{style="color:#E52212"} [**dsipersion**]{style="color:#E52212"}

Beta dispersion describes the spead of a set of points around a centroid

PERMANOVA can be sensitve to difference in beta-dispersion $so test for this too$

![](img/Diversity_workshop_JWforwebsite192.png)

# Binary transform -- why?

It tells us about the influence of rare members and abundance on beta-diversity scores

|       | Community A | Community B | Community C |
|:-----:|:-----------:|:-----------:|:-----------:|
| ESV 1 |     50      |     60      |     10      |
| ESV 2 |      5      |     10      |      2      |
| ESV 3 |      3      |      7      |      3      |
| ESV 4 |      0      |      0      |      1      |
| ESV 5 |      0      |      0      |      2      |

|       | Community A | Community B | Community C |
|:-----:|:-----------:|:-----------:|:-----------:|
| ESV 1 |      1      |      1      |      1      |
| ESV 2 |      1      |      1      |      1      |
| ESV 3 |      1      |      1      |      1      |
| ESV 4 |      0      |      0      |      1      |
| ESV 5 |      0      |      0      |      1      |

# Binary transformation helps determine why a community is (dis)similar

| Dominant species | Rare species | Weighted Dissimilarly | Binary transformed dissimilarity |     Dissimilarity due to (hypothesis):<br />     |
|:----------------:|:------------:|:---------------------:|:--------------------------------:|:------------------------------------------------:|
|       Same       |     Same     |          Low          |               Low                |   No replacement<br />Communities are the same   |
|       Same       |   Distinct   |          Low          |               High               | Partial Replacement <br />(convergent selection) |
|     Distinct     |     Same     |         High          |               Low                |    Partial Replacement (competitive turnover)    |
|     Distinct     |   Distinct   |         High          |               High               |     Full Replacement OR<br />High nestedness     |

Communities that share dominate species appear highly similar. \_\_ Q:\_\_ what happen when we binary transform?

Tank 1Richness = 3 \| Net abundance = 20

Tank 2Richness = 3 \| Net abundance = 20

![](img/Diversity_workshop_JWforwebsite193.png)

![](img/Diversity_workshop_JWforwebsite194.png)

![](img/Diversity_workshop_JWforwebsite195.png)

![](img/Diversity_workshop_JWforwebsite196.png)

![](img/Diversity_workshop_JWforwebsite197.png)

![](img/Diversity_workshop_JWforwebsite198.png)

![](img/Diversity_workshop_JWforwebsite199.png)

![](img/Diversity_workshop_JWforwebsite200.png)

![](img/Diversity_workshop_JWforwebsite201.png)

![](img/Diversity_workshop_JWforwebsite202.png)

![](img/Diversity_workshop_JWforwebsite203.png)

![](img/Diversity_workshop_JWforwebsite204.png)

![](img/Diversity_workshop_JWforwebsite205.png)

![](img/Diversity_workshop_JWforwebsite206.png)

![](img/Diversity_workshop_JWforwebsite207.png)

![](img/Diversity_workshop_JWforwebsite208.png)

![](img/Diversity_workshop_JWforwebsite209.png)

![](img/Diversity_workshop_JWforwebsite210.png)

![](img/Diversity_workshop_JWforwebsite211.png)

![](img/Diversity_workshop_JWforwebsite212.png)

![](img/Diversity_workshop_JWforwebsite213.png)

![](img/Diversity_workshop_JWforwebsite214.png)

![](img/Diversity_workshop_JWforwebsite215.png)

![](img/Diversity_workshop_JWforwebsite216.png)

![](img/Diversity_workshop_JWforwebsite217.png)

![](img/Diversity_workshop_JWforwebsite218.png)

![](img/Diversity_workshop_JWforwebsite219.png)

![](img/Diversity_workshop_JWforwebsite220.png)

![](img/Diversity_workshop_JWforwebsite221.png)

![](img/Diversity_workshop_JWforwebsite222.png)

![](img/Diversity_workshop_JWforwebsite223.png)

![](img/Diversity_workshop_JWforwebsite224.png)

![](img/Diversity_workshop_JWforwebsite225.png)

![](img/Diversity_workshop_JWforwebsite226.png)

![](img/Diversity_workshop_JWforwebsite227.png)

![](img/Diversity_workshop_JWforwebsite228.png)

![](img/Diversity_workshop_JWforwebsite229.png)

![](img/Diversity_workshop_JWforwebsite230.png)

![](img/Diversity_workshop_JWforwebsite231.png)

![](img/Diversity_workshop_JWforwebsite232.png)

![](img/Diversity_workshop_JWforwebsite233.png)

![](img/Diversity_workshop_JWforwebsite234.png)

![](img/Diversity_workshop_JWforwebsite235.png)

![](img/Diversity_workshop_JWforwebsite236.png)

![](img/Diversity_workshop_JWforwebsite237.png)

2$0 \+ 0 \+ 0 \+ 16 \+ 0$ = 32 = 0.8

20 + 20 40

**Q:** What happen when we binary transform? **A:** Any differences in rare community members will cause communities to look less similar $i\.e\. increase dissimilarity$

Tank 1Richness = 3 \| Net abundance = 3

Tank 2Richness = 3 \| Net abundance = 3

![](img/Diversity_workshop_JWforwebsite238.png)

![](img/Diversity_workshop_JWforwebsite239.png)

![](img/Diversity_workshop_JWforwebsite240.png)

![](img/Diversity_workshop_JWforwebsite241.png)

![](img/Diversity_workshop_JWforwebsite242.png)

![](img/Diversity_workshop_JWforwebsite243.png)

![](img/Diversity_workshop_JWforwebsite244.png)

![](img/Diversity_workshop_JWforwebsite245.png)

![](img/Diversity_workshop_JWforwebsite246.png)

![](img/Diversity_workshop_JWforwebsite247.png)

![](img/Diversity_workshop_JWforwebsite248.png)

2$0 \+ 0 \+ 0 \+ 1 \+ 0$ = 2 = 0.3

3 + 3 6

The communities with identical richness and biomass are \>50% dissimilar **due dominant species turnover.** **Q:** what happens when we binary transform?

Tank 2Richness = 5 \| Net abundance = 16

Tank 1Richness = 5 \| Net abundance = 16

![](img/Diversity_workshop_JWforwebsite249.png)

![](img/Diversity_workshop_JWforwebsite250.png)

![](img/Diversity_workshop_JWforwebsite251.png)

![](img/Diversity_workshop_JWforwebsite252.png)

![](img/Diversity_workshop_JWforwebsite253.png)

![](img/Diversity_workshop_JWforwebsite254.png)

![](img/Diversity_workshop_JWforwebsite255.png)

![](img/Diversity_workshop_JWforwebsite256.png)

![](img/Diversity_workshop_JWforwebsite257.png)

![](img/Diversity_workshop_JWforwebsite258.png)

![](img/Diversity_workshop_JWforwebsite259.png)

![](img/Diversity_workshop_JWforwebsite260.png)

![](img/Diversity_workshop_JWforwebsite261.png)

![](img/Diversity_workshop_JWforwebsite262.png)

![](img/Diversity_workshop_JWforwebsite263.png)

![](img/Diversity_workshop_JWforwebsite264.png)

![](img/Diversity_workshop_JWforwebsite265.png)

![](img/Diversity_workshop_JWforwebsite266.png)

![](img/Diversity_workshop_JWforwebsite267.png)

![](img/Diversity_workshop_JWforwebsite268.png)

![](img/Diversity_workshop_JWforwebsite269.png)

![](img/Diversity_workshop_JWforwebsite270.png)

![](img/Diversity_workshop_JWforwebsite271.png)

![](img/Diversity_workshop_JWforwebsite272.png)

![](img/Diversity_workshop_JWforwebsite273.png)

![](img/Diversity_workshop_JWforwebsite274.png)

![](img/Diversity_workshop_JWforwebsite275.png)

![](img/Diversity_workshop_JWforwebsite276.png)

![](img/Diversity_workshop_JWforwebsite277.png)

![](img/Diversity_workshop_JWforwebsite278.png)

![](img/Diversity_workshop_JWforwebsite279.png)

![](img/Diversity_workshop_JWforwebsite280.png)

![](img/Diversity_workshop_JWforwebsite281.png)

![](img/Diversity_workshop_JWforwebsite282.png)

![](img/Diversity_workshop_JWforwebsite283.png)

![](img/Diversity_workshop_JWforwebsite284.png)

2$2 \+ 1 \+ 1 \+ 2 \+ 0$ = 12 = 0.39

1 - 0.39 = 0.61

13 + 14 31

**Q:** What happen when we binary transform? **A:** Any differences in rare community members will cause communities to look more similar $i\.e\. __decrease__ dissimilarity$

Tank 1Richness = 5 \| Net abundance = 5

Tank 2Richness = 5 \| Net abundance = 5

![](img/Diversity_workshop_JWforwebsite285.png)

![](img/Diversity_workshop_JWforwebsite286.png)

![](img/Diversity_workshop_JWforwebsite287.png)

![](img/Diversity_workshop_JWforwebsite288.png)

![](img/Diversity_workshop_JWforwebsite289.png)

![](img/Diversity_workshop_JWforwebsite290.png)

![](img/Diversity_workshop_JWforwebsite291.png)

![](img/Diversity_workshop_JWforwebsite292.png)

![](img/Diversity_workshop_JWforwebsite293.png)

![](img/Diversity_workshop_JWforwebsite294.png)

![](img/Diversity_workshop_JWforwebsite295.png)

![](img/Diversity_workshop_JWforwebsite296.png)

![](img/Diversity_workshop_JWforwebsite297.png)

![](img/Diversity_workshop_JWforwebsite298.png)

2$1 \+ 1 \+ 1 \+ 1 \+ 0$ = 8 = 0.8

5 + 5 10

# Why do I need to normalise?

-   It is very difficult to collect exactly the same number of sequence reads for each sample
    -   There is an arbitrary total read depth imposed by the instrument
    -   Major confounder for distance or dissimilarity calculations and multivariate ordinations derived from these distances

![](img/Diversity_workshop_JWforwebsite299.png)

Library size $reads$

[**What is rarefying? Why rarefy?**]{style="color:#C00000"}

**"rarefy": a normalization technique** $__ _rarefying_ __$

\_ \_ Not to be confused with "rarefaction": a technique for species richness estimation

Rarefying aims to reduce bias introduced by variable library sizes

![](img/Diversity_workshop_JWforwebsite300.png)

Rarefying involves choosing a threshold of reads to which to normalize all your samples to

$e\.g\. 20\,000\, min sample read depth$

Samples that fall below the threshold will be removed from your analysis

The goal is to choose a threshold that keeps the most reads whilst also retaining the most samples

Library size $reads$

Too high. Many samples lost

Too low? Many reads lost

# Normalization issues -- why not rarefy?

Subsampling results in a loss of information and precision

Estimated normalization factors should ensure that an ESV with the same abundance in two samples is not detected as differentially expressed $DE$

*Sample A*

*ESV* \_ count\_

1 20

2 20

3 20

4 20

5 20

*6 0\_*

Total 100

*Sample B*

*ESV* \_ count\_

1 20

2 20

3 20

4 20

5 20

\_ 6 100\_

Total 200

*Sample A*

*ESV* \_ count\_

1 20

2 20

3 20

4 20

5 20

*6 0\_*

Total 100

*Sample B*

*ESV* \_ count\_

1 10

2 10

3 10

4 10

5 10

*6 50*

Total 100

[What if OTU 6 is chloroplast DNA and samples are bulk and rhizosphere soil??]{style="color:#C00000"}

Sample to a set depth $100$

Everything will show up as differentially expressed!

McMurdie PJ, Holmes S. 2014. Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible. PLoS Comput Biol 10$4$.

# Other normalisation options

Total sum scaling $TSS$

Converts a community to proportions

No information is lost

Compositional nature of NGS data can confound analyses

![](img/Diversity_workshop_JWforwebsite301.png)

![](img/Diversity_workshop_JWforwebsite302.png)

-   Centre log-ratio $CLR$ transformation
-   CLR for an entry 'xi' = Log$xi/gm\(x$), where gm$x$ is the geometric mean of all entries for ESV 'x'
    -   Gm for 2 number is square of their product, gm of 3 number is cube root of their product
-   CLR transformed values are scale-invariant. Same ratio $xi/gm\(x$ is expected to be obtained in a sample with few read counts or an identical sample with many read counts, only the precision of the CLR estimate is affected
    -   Designed to manage compositional data
-   Geometric mean applies only to positive numbers $so zeros need to be managed$
    -   ALDEx2, robust CLR

# Metrics: defining 'distance'

-   Are a measure of distance, in this case between two communities similarity
-   Euclidian metric
    -   Biases single large differences
-   Manhattan
    -   No upper limit
-   Bray-Curtis metric

![](img/Diversity_workshop_JWforwebsite303.png)

![](img/Diversity_workshop_JWforwebsite304.png)

![](img/Diversity_workshop_JWforwebsite305.png)

# Moving beyond Bray-Curtis

Bray-Curtis index $for compositional data$

Jaccard index $for presence / absence data\, ignores abundance information$

Aitchison distance $Euclidean distance for CLR transformed abundances\, aiming to avoid the compositionality bias$

Unifrac distances $that take into account phylogenetic relationships$

*Community C*

*Pseudomonas fluorescens*

*Pseudomonas putida*

*Pseudomonas aeruginosa*

*Community A*

*Bacillus cereus*

*Bacillus anthracis*

*Bacillus* *thuringensis*

[**Community B**]{style="color:#51A294"}

[Pseudomonas]{style="color:#51A294"} [mendocina]{style="color:#51A294"}

[Pseudomonas]{style="color:#51A294"} [resinovorans]{style="color:#51A294"}

[Pseudomonas]{style="color:#51A294"} [veronii]{style="color:#51A294"}

# Further reading

-   Knight R, Vrbanac AF, Taylor BC, Aksenov A, Callewaert C, Debelius JW, Gonzalez A, Kosciólek T, McCall L-I, McDonald D, Melnik A, Morton J, Navas J, Quinn R, Sanders J, Swafford AD, Thompson L, Tripathi A, Xu Z, Zaneveld J, Zhu Q, Caporaso J, Dorrestein P. 2018. Best practices for analysing microbiomes. Nat Rev Microbiol 16$410\-422$.
-   Gloor GB, Macklaim JM, Pawlowsky-Glahn V, Egozcue JJ. 2017. Microbiome Datasets Are Compositional: And This Is Not Optional. Front Microbiol 8$2224\-2224$
-   McMurdie PJ, Holmes S. 2014. Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible. PLoS Comput Biol 10$4$.
-   Lin H, Peddada SD. 2020. Analysis of compositions of microbiomes with bias correction. Nat Commun 11$1$:3514.
-   Ellen S. Camerona PJS, Benjamin J.M. Tremblaya, Monica B. Emelkob, Kirsten M. Müller. 2020. To rarefy or not to rarefy: Enhancing microbial community analysis through nextgeneration sequencing.

# Further resources

[*https://sites.google.com/site/mb3gustame/home*](https://sites.google.com/site/mb3gustame/home) \_ \_ Guide to major multivariate test with explanations and walkthroughs using published papers

<https://youtu.be/j1IbfQrT2Cs> Compositional Data Analysis Approaches to Improve Microbiome Studies: from Collection to Conclusions
