<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Plotting diversity | Workflows for processing microbial amplicon sequences</title>
  <meta name="description" content="This GitBook contains bioinformatic workflows from raw microbial amplicon-sequence reads generated with a Miseq instrument to pre-processing reads with packages such as Qiime2, followed by applying various packages within the R Environment." />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Plotting diversity | Workflows for processing microbial amplicon sequences" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This GitBook contains bioinformatic workflows from raw microbial amplicon-sequence reads generated with a Miseq instrument to pre-processing reads with packages such as Qiime2, followed by applying various packages within the R Environment." />
  <meta name="github-repo" content="chrismitbiz/ABlab-workflows" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Plotting diversity | Workflows for processing microbial amplicon sequences" />
  
  <meta name="twitter:description" content="This GitBook contains bioinformatic workflows from raw microbial amplicon-sequence reads generated with a Miseq instrument to pre-processing reads with packages such as Qiime2, followed by applying various packages within the R Environment." />
  

<meta name="author" content="Christian Krohn, PhD, RMIT University" />


<meta name="date" content="2022-07-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Qiime2R.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A GitBook for Teaching</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this GitBook</a></li>
<li class="chapter" data-level="2" data-path="gettingstarted.html"><a href="gettingstarted.html"><i class="fa fa-check"></i><b>2</b> Getting started</a></li>
<li class="chapter" data-level="3" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html"><i class="fa fa-check"></i><b>3</b> Miseq library preps</a>
<ul>
<li class="chapter" data-level="3.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#overview"><i class="fa fa-check"></i><b>3.1.1</b> Overview</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#workflow"><i class="fa fa-check"></i><b>3.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-extraction"><i class="fa fa-check"></i><b>3.2.1</b> DNA extraction</a></li>
<li class="chapter" data-level="3.2.2" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-quality-and-concentration"><i class="fa fa-check"></i><b>3.2.2</b> DNA quality and concentration</a></li>
<li class="chapter" data-level="3.2.3" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-normalisation"><i class="fa fa-check"></i><b>3.2.3</b> DNA normalisation</a></li>
<li class="chapter" data-level="3.2.4" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#first-and-second-pcr---amplicon-pcr-and-indexing-pcr"><i class="fa fa-check"></i><b>3.2.4</b> First and second PCR - Amplicon PCR and Indexing PCR</a></li>
<li class="chapter" data-level="3.2.5" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#normalisation-of-indexed-amplicons"><i class="fa fa-check"></i><b>3.2.5</b> Normalisation of indexed amplicons</a></li>
<li class="chapter" data-level="3.2.6" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#pooling"><i class="fa fa-check"></i><b>3.2.6</b> Pooling</a></li>
<li class="chapter" data-level="3.2.7" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#denaturing-of-pool-and-loading-of-the-miseq"><i class="fa fa-check"></i><b>3.2.7</b> Denaturing of pool and loading of the Miseq</a></li>
<li class="chapter" data-level="3.2.8" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#consumables"><i class="fa fa-check"></i><b>3.2.8</b> Consumables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html"><i class="fa fa-check"></i><b>4</b> From BaseSpace to Qiime2 and DADA2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#prerequisites-and-required-files"><i class="fa fa-check"></i><b>4.1.1</b> Prerequisites and required files</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#workflow-1"><i class="fa fa-check"></i><b>4.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#download-fastq-files"><i class="fa fa-check"></i><b>4.2.1</b> Download FASTQ files</a></li>
<li class="chapter" data-level="4.2.2" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#paired-end-manifest-import-step-1"><i class="fa fa-check"></i><b>4.2.2</b> Paired end manifest import (Step 1)</a></li>
<li class="chapter" data-level="4.2.3" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#cutadapt-step-2"><i class="fa fa-check"></i><b>4.2.3</b> Cutadapt (Step 2)</a></li>
<li class="chapter" data-level="4.2.4" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#read-quality-assessment"><i class="fa fa-check"></i><b>4.2.4</b> Read quality assessment</a></li>
<li class="chapter" data-level="4.2.5" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#denoise-paired-end-sequences-with-dada2-step-3"><i class="fa fa-check"></i><b>4.2.5</b> Denoise paired end sequences with dada2 (Step 3)</a></li>
<li class="chapter" data-level="4.2.6" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#taxonomic-classifier-and-assignment-step-4"><i class="fa fa-check"></i><b>4.2.6</b> Taxonomic classifier and assignment (Step 4)</a></li>
<li class="chapter" data-level="4.2.7" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#build-phylogenetic-tree-step-5"><i class="fa fa-check"></i><b>4.2.7</b> Build phylogenetic tree (Step 5)</a></li>
<li class="chapter" data-level="4.2.8" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#all-steps-combined"><i class="fa fa-check"></i><b>4.2.8</b> All steps combined</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Qiime2R.html"><a href="Qiime2R.html"><i class="fa fa-check"></i><b>5</b> From Qiime2 into R - Initial diagnostics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Qiime2R.html"><a href="Qiime2R.html#introduction-2"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="Qiime2R.html"><a href="Qiime2R.html#workflow-2"><i class="fa fa-check"></i><b>5.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Qiime2R.html"><a href="Qiime2R.html#packages"><i class="fa fa-check"></i><b>5.2.1</b> Packages</a></li>
<li class="chapter" data-level="5.2.2" data-path="Qiime2R.html"><a href="Qiime2R.html#import-qiime-files-and-create-a-phyloseq-object"><i class="fa fa-check"></i><b>5.2.2</b> Import qiime-files and create a phyloseq object</a></li>
<li class="chapter" data-level="5.2.3" data-path="Qiime2R.html"><a href="Qiime2R.html#save-the-ps-object-as-an-.rds-file-into-your-working-directory"><i class="fa fa-check"></i><b>5.2.3</b> Save the <code>ps</code> object as an <code>.rds</code> file into your working directory</a></li>
<li class="chapter" data-level="5.2.4" data-path="Qiime2R.html"><a href="Qiime2R.html#initial-filtering"><i class="fa fa-check"></i><b>5.2.4</b> Initial filtering</a></li>
<li class="chapter" data-level="5.2.5" data-path="Qiime2R.html"><a href="Qiime2R.html#rarefaction-curve"><i class="fa fa-check"></i><b>5.2.5</b> Rarefaction curve</a></li>
<li class="chapter" data-level="5.2.6" data-path="Qiime2R.html"><a href="Qiime2R.html#prevalence-table"><i class="fa fa-check"></i><b>5.2.6</b> Prevalence table</a></li>
<li class="chapter" data-level="5.2.7" data-path="Qiime2R.html"><a href="Qiime2R.html#export-asv-and-taxa-tables-to-excel"><i class="fa fa-check"></i><b>5.2.7</b> Export ASV and taxa tables to Excel</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="plotting-diversity.html"><a href="plotting-diversity.html"><i class="fa fa-check"></i><b>6</b> Plotting diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="plotting-diversity.html"><a href="plotting-diversity.html#introduction-3"><i class="fa fa-check"></i><b>6.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="plotting-diversity.html"><a href="plotting-diversity.html#prerequisites"><i class="fa fa-check"></i><b>6.1.1</b> Prerequisites</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="plotting-diversity.html"><a href="plotting-diversity.html#workflow-3"><i class="fa fa-check"></i><b>6.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="plotting-diversity.html"><a href="plotting-diversity.html#packages-1"><i class="fa fa-check"></i><b>6.2.1</b> Packages</a></li>
<li class="chapter" data-level="6.2.2" data-path="plotting-diversity.html"><a href="plotting-diversity.html#load-phyloseq-object"><i class="fa fa-check"></i><b>6.2.2</b> Load phyloseq object</a></li>
<li class="chapter" data-level="6.2.3" data-path="plotting-diversity.html"><a href="plotting-diversity.html#pre-filter-rarefy-and-calculate-diversity"><i class="fa fa-check"></i><b>6.2.3</b> Pre-filter, rarefy and calculate diversity</a></li>
<li class="chapter" data-level="6.2.4" data-path="plotting-diversity.html"><a href="plotting-diversity.html#colors"><i class="fa fa-check"></i><b>6.2.4</b> Colors</a></li>
<li class="chapter" data-level="6.2.5" data-path="plotting-diversity.html"><a href="plotting-diversity.html#richness"><i class="fa fa-check"></i><b>6.2.5</b> Richness</a></li>
<li class="chapter" data-level="6.2.6" data-path="plotting-diversity.html"><a href="plotting-diversity.html#shannon"><i class="fa fa-check"></i><b>6.2.6</b> Shannon</a></li>
<li class="chapter" data-level="6.2.7" data-path="plotting-diversity.html"><a href="plotting-diversity.html#pielous-evenness"><i class="fa fa-check"></i><b>6.2.7</b> Pielou’s evenness</a></li>
<li class="chapter" data-level="6.2.8" data-path="plotting-diversity.html"><a href="plotting-diversity.html#combined-plotting"><i class="fa fa-check"></i><b>6.2.8</b> Combined plotting</a></li>
<li class="chapter" data-level="6.2.9" data-path="plotting-diversity.html"><a href="plotting-diversity.html#nmds"><i class="fa fa-check"></i><b>6.2.9</b> NMDS</a></li>
<li class="chapter" data-level="6.2.10" data-path="plotting-diversity.html"><a href="plotting-diversity.html#pca"><i class="fa fa-check"></i><b>6.2.10</b> PCA</a></li>
<li class="chapter" data-level="6.2.11" data-path="plotting-diversity.html"><a href="plotting-diversity.html#rda"><i class="fa fa-check"></i><b>6.2.11</b> RDA</a></li>
<li class="chapter" data-level="6.2.12" data-path="plotting-diversity.html"><a href="plotting-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>6.2.12</b> Unweighted uniFrac</a></li>
<li class="chapter" data-level="6.2.13" data-path="plotting-diversity.html"><a href="plotting-diversity.html#combined-plotting-1"><i class="fa fa-check"></i><b>6.2.13</b> Combined plotting</a></li>
<li class="chapter" data-level="6.2.14" data-path="plotting-diversity.html"><a href="plotting-diversity.html#save-plots-to-png-and-pdf"><i class="fa fa-check"></i><b>6.2.14</b> Save plots to png and pdf</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workflows for processing microbial amplicon sequences</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="plotting-diversity" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Plotting diversity<a href="plotting-diversity.html#plotting-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Some visualisations of alpha and beta diversity.</p>
<div id="introduction-3" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Introduction<a href="plotting-diversity.html#introduction-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This chapter introduces you to a basic workflow for plotting alpha and beta diversity using phyloseq and ggplot. It provides examples for loading the phyloseq object from a previously saved .rds file, then how to calculate and include alpha diversity metrics into the sample_data of the phyloseq object, and finally how to plot some of these metrics, individually or combined.</p>
<p>Furthermore, beta diversity, i.e. four different type of ordinations are plotted. The examples, include non-metric dimensional scaling on Bray-Curtis dissimilaries, Principal Component Analysis (Aitchison distance), Redundancy Analysis ( Aitchison) and a Principal Coordinate Analysis on unweighted uniFrac distances.</p>
<p>The stats behind the individual metrics are not discussed. If you want to learn more about the various options to analyse your amplicon-derived diversity, perhaps start here: <a href="https://sites.google.com/site/mb3gustame/home" class="uri">https://sites.google.com/site/mb3gustame/home</a>, the GUide to STatistical Analysis in Microbial Ecology (GUSTA ME). This is one of many other useful links and resources. Some others are listed in chapter <a href="index.html#about">1</a>.</p>
<p>Prior to establishing alpha diversity indices, as well as Bray-Curtis dissimilarities, we will stick to rarefying sequences to a specific library size (library size = sum of sequence counts of a sample) to normalise the data and prevent bias due to library size. Usually, this is done using the smallest library size of all samples. If you want to learn more about rarefying I recommend this paper here: <a href="https://www.nature.com/articles/s41598-021-01636-1">Enhancing diversity analysis by repeatedly rarefying next generation sequencing data describing microbial communities</a>. We will follow the recommendations from the authors of this paper to resample without replacement.</p>
<p>Other normalisation methods are also possible and it is up to you, the analyst, to understand and decide on the appropriate methods. I found centred-log ratios (Aitchison distances) also intuitive in interpretation, which are included into the workflow below in a PCA and RDA. Log-ratios are further discussed in the paper <a href="https://doi.org/10.3389/fmicb.2017.02224">Microbiome Datasets Are Compositional: And This Is Not Optional</a>.<br />
In fact, using different types of normalisation, dissimilarities or distances, while providing different types of information, may complement each other and help in interpretation of your data.</p>
<div id="prerequisites" class="section level3 hasAnchor" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Prerequisites<a href="plotting-diversity.html#prerequisites" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>a phyloseq object; either read in from a pre-saved <code>.rds</code> file or created as described in chapter <a href="Qiime2R.html#Qiime2R">5</a>.</li>
</ul>
<p>Time to get started…</p>
</div>
</div>
<div id="workflow-3" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Workflow<a href="plotting-diversity.html#workflow-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="packages-1" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Packages<a href="plotting-diversity.html#packages-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Install the following packages if not already.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="plotting-diversity.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install(&quot;microbiome&quot;) If needed</span></span>
<span id="cb28-2"><a href="plotting-diversity.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb28-3"><a href="plotting-diversity.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)       <span class="co"># a handy helper package for ggplots</span></span>
<span id="cb28-4"><a href="plotting-diversity.html#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb28-5"><a href="plotting-diversity.html#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())  <span class="co"># setting the theme for ggplots</span></span>
<span id="cb28-6"><a href="plotting-diversity.html#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb28-7"><a href="plotting-diversity.html#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span></code></pre></div>
<p><br />
</p>
</div>
<div id="load-phyloseq-object" class="section level3 hasAnchor" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Load phyloseq object<a href="plotting-diversity.html#load-phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For detail on how to create a phyloseq object please see chapter <a href="Qiime2R.html#Qiime2R">5</a>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="plotting-diversity.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reading in a previously saved phyloseq object</span></span>
<span id="cb29-2"><a href="plotting-diversity.html#cb29-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&#39;ps_ProjectX_2022July&#39;</span>)</span>
<span id="cb29-3"><a href="plotting-diversity.html#cb29-3" aria-hidden="true" tabindex="-1"></a>ps  <span class="co"># get an overview of number of taxa and samples contained in the phyloseq object</span></span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 4218 taxa and 51 samples ]
## sample_data() Sample Data:       [ 51 samples by 15 sample variables ]
## tax_table()   Taxonomy Table:    [ 4218 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 4218 tips and 4217 internal nodes ]</code></pre>
<p><br />
</p>
<p>Check the metadata contained in the phyloseq object. This is the original metadata from the public repository where the FastQ files were downloaded from (Sequence Read Archives).</p>
<p>There are 15 columns containing some character (chr), factor or integer (int) variables.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="plotting-diversity.html#cb31-1" aria-hidden="true" tabindex="-1"></a>metadf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps))</span>
<span id="cb31-2"><a href="plotting-diversity.html#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(metadf)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    51 obs. of  15 variables:
##  $ Assay.Type     : chr  &quot;AMPLICON&quot; &quot;AMPLICON&quot; &quot;AMPLICON&quot; &quot;AMPLICON&quot; ...
##  $ Bases          : int  13254234 24196788 15758554 20048406 37634632 43555904 35107436 13888140 23172184 35065898 ...
##  $ BioProject     : chr  &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; ...
##  $ Bytes          : int  9235913 16404070 10758456 13009806 24207581 27963491 22585685 9388862 15125835 22520074 ...
##  $ Organism       : chr  &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; ...
##  $ collection_date: chr  &quot;30/11/16&quot; &quot;9/3/16&quot; &quot;17/3/16&quot; &quot;5/12/16&quot; ...
##  $ loc            : chr  &quot;Ejby Moelle&quot; &quot;Soeholt&quot; &quot;Esbjerg Vest&quot; &quot;Viborg&quot; ...
##  $ Reactor        : Factor w/ 10 levels &quot;1A&quot;,&quot;1B&quot;,&quot;2B&quot;,..: 4 4 6 7 7 5 5 5 4 4 ...
##  $ Type           : chr  NA NA NA &quot;DS&quot; ...
##  $ Purpose        : chr  &quot;reactorfoampotentials&quot; &quot;reactorfoampotentials&quot; &quot;reactorfoampotentials&quot; &quot;foamnofoamcomparison&quot; ...
##  $ Experiment     : chr  &quot;SRX8715180&quot; &quot;SRX8715169&quot; &quot;SRX8715158&quot; &quot;SRX8715151&quot; ...
##  $ lat_lon        : chr  &quot;55.398077 N 10.415041 E&quot; &quot;56.175302 N 9.582588 E&quot; &quot;55.488235 N 8.430655 E&quot; &quot;56.425367 N 9.4527943 E&quot; ...
##  $ ReleaseDate    : chr  &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; ...
##  $ Sample.Name    : chr  &quot;16SAMP-17236&quot; &quot;16SAMP-17231&quot; &quot;16SAMP-17233&quot; &quot;MQ170915-14&quot; ...
##  $ Location       : Factor w/ 12 levels &quot;Avedoere&quot;,&quot;Damhusaaen&quot;,..: 4 11 5 12 12 12 12 2 11 11 ...</code></pre>
<p><br />
</p>
</div>
<div id="pre-filter-rarefy-and-calculate-diversity" class="section level3 hasAnchor" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Pre-filter, rarefy and calculate diversity<a href="plotting-diversity.html#pre-filter-rarefy-and-calculate-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now the diversity indices can be calculated and added to the metadata of the phyloseq object for plotting. First, the abundances are pre-filtered (as shown in the previous chapter), then the ASV abundances are randomly resampled with <code>phyloseq::rarefy_even_depth</code>, such that all samples have the same number of ASVs.<br />
The diversity indices are then calculated with <code>phyloseq::estimate_richness</code> based on these evenly resampled abundances. The dataframe that is created containing the range of indices is then added to the existing metadata of the phyloseq for plotting.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="plotting-diversity.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># some additional quality filtering - repeating steps from previous chapter. </span></span>
<span id="cb33-2"><a href="plotting-diversity.html#cb33-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb33-3"><a href="plotting-diversity.html#cb33-3" aria-hidden="true" tabindex="-1"></a>ps.flt <span class="ot">=</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps) <span class="sc">&gt;=</span> <span class="dv">5</span>, ps) <span class="co">#minimum reads per feature </span></span>
<span id="cb33-4"><a href="plotting-diversity.html#cb33-4" aria-hidden="true" tabindex="-1"></a>ps.flt  <span class="ot">=</span> <span class="fu">subset_taxa</span>(ps.flt , <span class="sc">!</span><span class="fu">is.na</span>(Phylum) <span class="sc">&amp;</span> <span class="sc">!</span>Phylum <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;&quot;</span>))</span>
<span id="cb33-5"><a href="plotting-diversity.html#cb33-5" aria-hidden="true" tabindex="-1"></a>ps.flt <span class="ot">&lt;-</span> ps.flt <span class="sc">%&gt;%</span></span>
<span id="cb33-6"><a href="plotting-diversity.html#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_taxa</span>(Kingdom <span class="sc">==</span> <span class="st">&quot;Bacteria&quot;</span> <span class="sc">&amp;</span> Family  <span class="sc">!=</span> <span class="st">&quot;Mitochondria&quot;</span> <span class="sc">&amp;</span> Class   <span class="sc">!=</span> <span class="st">&quot;Chloroplast&quot;</span>)</span>
<span id="cb33-7"><a href="plotting-diversity.html#cb33-7" aria-hidden="true" tabindex="-1"></a>ps.flt </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 3853 taxa and 51 samples ]
## sample_data() Sample Data:       [ 51 samples by 15 sample variables ]
## tax_table()   Taxonomy Table:    [ 3853 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 3853 tips and 3852 internal nodes ]</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="plotting-diversity.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># confirming minimum sample size</span></span>
<span id="cb35-2"><a href="plotting-diversity.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">min</span>(<span class="fu">colSums</span>(<span class="fu">otu_table</span>(ps.flt)))</span></code></pre></div>
<pre><code>## [1] 9770</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="plotting-diversity.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rarefy - required before measuring alpha diversity. </span></span>
<span id="cb37-2"><a href="plotting-diversity.html#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This step is to create an abundance table, where the all samples are randomly </span></span>
<span id="cb37-3"><a href="plotting-diversity.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co"># resamples such that all samples have the same number of ASVs.  </span></span>
<span id="cb37-4"><a href="plotting-diversity.html#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co"># It removes any bias but number of re-samples is based on the sample with </span></span>
<span id="cb37-5"><a href="plotting-diversity.html#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co"># smallest number of ASVs so one has to check and filter beforehand in case </span></span>
<span id="cb37-6"><a href="plotting-diversity.html#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co"># there are samples with very different number of ASVs. I am generally going </span></span>
<span id="cb37-7"><a href="plotting-diversity.html#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co"># by a 10x rule. If the lowest number of ASVs is 10x lower than the sample with</span></span>
<span id="cb37-8"><a href="plotting-diversity.html#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co"># highest number of ASVs, then I keep that sample in for alpha diversity assessments.  </span></span>
<span id="cb37-9"><a href="plotting-diversity.html#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="plotting-diversity.html#cb37-10" aria-hidden="true" tabindex="-1"></a>ps_rare <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">rarefy_even_depth</span>(ps.flt, </span>
<span id="cb37-11"><a href="plotting-diversity.html#cb37-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">sample.size =</span> <span class="fu">min</span>(<span class="fu">colSums</span>(<span class="fu">otu_table</span>(ps.flt))),  </span>
<span id="cb37-12"><a href="plotting-diversity.html#cb37-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">rngseed =</span> <span class="cn">TRUE</span>, </span>
<span id="cb37-13"><a href="plotting-diversity.html#cb37-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="co">#without replacement</span></span>
<span id="cb37-14"><a href="plotting-diversity.html#cb37-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">trimOTUs =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-15"><a href="plotting-diversity.html#cb37-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-16"><a href="plotting-diversity.html#cb37-16" aria-hidden="true" tabindex="-1"></a><span class="co"># check rarefied phyloseq object</span></span>
<span id="cb37-17"><a href="plotting-diversity.html#cb37-17" aria-hidden="true" tabindex="-1"></a><span class="co"># ps_rare (not run)</span></span>
<span id="cb37-18"><a href="plotting-diversity.html#cb37-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-19"><a href="plotting-diversity.html#cb37-19" aria-hidden="true" tabindex="-1"></a><span class="co"># phyloseq-class experiment-level object</span></span>
<span id="cb37-20"><a href="plotting-diversity.html#cb37-20" aria-hidden="true" tabindex="-1"></a><span class="co"># otu_table()   OTU Table:         [ 3801 taxa and 51 samples ]</span></span>
<span id="cb37-21"><a href="plotting-diversity.html#cb37-21" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_data() Sample Data:       [ 51 samples by 15 sample variables ]</span></span>
<span id="cb37-22"><a href="plotting-diversity.html#cb37-22" aria-hidden="true" tabindex="-1"></a><span class="co"># tax_table()   Taxonomy Table:    [ 3801 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb37-23"><a href="plotting-diversity.html#cb37-23" aria-hidden="true" tabindex="-1"></a> <span class="co">#phy_tree()    Phylogenetic Tree: [ 3801 tips and 3800 internal nodes ]</span></span>
<span id="cb37-24"><a href="plotting-diversity.html#cb37-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-25"><a href="plotting-diversity.html#cb37-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Create diversity indices</span></span>
<span id="cb37-26"><a href="plotting-diversity.html#cb37-26" aria-hidden="true" tabindex="-1"></a>div.df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">estimate_richness</span>(ps_rare)</span>
<span id="cb37-27"><a href="plotting-diversity.html#cb37-27" aria-hidden="true" tabindex="-1"></a><span class="co"># add sample names</span></span>
<span id="cb37-28"><a href="plotting-diversity.html#cb37-28" aria-hidden="true" tabindex="-1"></a>div.df<span class="sc">$</span><span class="st">`</span><span class="at">#SampleID</span><span class="st">`</span> <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">sample_names</span>(ps.flt)</span>
<span id="cb37-29"><a href="plotting-diversity.html#cb37-29" aria-hidden="true" tabindex="-1"></a><span class="co"># add diversity indices to phyloseq object</span></span>
<span id="cb37-30"><a href="plotting-diversity.html#cb37-30" aria-hidden="true" tabindex="-1"></a>metadf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps.flt))  <span class="co"># export metadata first</span></span>
<span id="cb37-31"><a href="plotting-diversity.html#cb37-31" aria-hidden="true" tabindex="-1"></a>metadf <span class="ot">&lt;-</span> metadf <span class="sc">%&gt;%</span> </span>
<span id="cb37-32"><a href="plotting-diversity.html#cb37-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;#SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-33"><a href="plotting-diversity.html#cb37-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(div.df, <span class="at">by =</span> <span class="st">&quot;#SampleID&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-34"><a href="plotting-diversity.html#cb37-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;#SampleID&quot;</span>) <span class="co"># add the diversity indices (columns) to the metadata</span></span>
<span id="cb37-35"><a href="plotting-diversity.html#cb37-35" aria-hidden="true" tabindex="-1"></a><span class="co"># add pilou evenness (see calculation in &quot;Numerical Ecology with R - Bocard et al&quot;)</span></span>
<span id="cb37-36"><a href="plotting-diversity.html#cb37-36" aria-hidden="true" tabindex="-1"></a>metadf<span class="sc">$</span>pielou_ev <span class="ot">&lt;-</span> div.df<span class="sc">$</span>Shannon<span class="sc">/</span><span class="fu">log</span>(div.df<span class="sc">$</span>Observed)</span>
<span id="cb37-37"><a href="plotting-diversity.html#cb37-37" aria-hidden="true" tabindex="-1"></a><span class="co"># replace metadata with temp meta.df which includes the diversity indices</span></span>
<span id="cb37-38"><a href="plotting-diversity.html#cb37-38" aria-hidden="true" tabindex="-1"></a>ps.flt<span class="sc">@</span>sam_data <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(metadf)  </span>
<span id="cb37-39"><a href="plotting-diversity.html#cb37-39" aria-hidden="true" tabindex="-1"></a>ps_rare<span class="sc">@</span>sam_data <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(metadf)  </span>
<span id="cb37-40"><a href="plotting-diversity.html#cb37-40" aria-hidden="true" tabindex="-1"></a><span class="co"># this replaced the old metadata with new metadata that includes the diversity </span></span>
<span id="cb37-41"><a href="plotting-diversity.html#cb37-41" aria-hidden="true" tabindex="-1"></a><span class="co"># indices for this phyloseq object </span></span>
<span id="cb37-42"><a href="plotting-diversity.html#cb37-42" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(<span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps.flt)))</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    51 obs. of  25 variables:
##  $ Assay.Type     : chr  &quot;AMPLICON&quot; &quot;AMPLICON&quot; &quot;AMPLICON&quot; &quot;AMPLICON&quot; ...
##  $ Bases          : int  13254234 24196788 15758554 20048406 37634632 43555904 35107436 13888140 23172184 35065898 ...
##  $ BioProject     : chr  &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; ...
##  $ Bytes          : int  9235913 16404070 10758456 13009806 24207581 27963491 22585685 9388862 15125835 22520074 ...
##  $ Organism       : chr  &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; ...
##  $ collection_date: chr  &quot;30/11/16&quot; &quot;9/3/16&quot; &quot;17/3/16&quot; &quot;5/12/16&quot; ...
##  $ loc            : chr  &quot;Ejby Moelle&quot; &quot;Soeholt&quot; &quot;Esbjerg Vest&quot; &quot;Viborg&quot; ...
##  $ Reactor        : Factor w/ 10 levels &quot;1A&quot;,&quot;1B&quot;,&quot;2B&quot;,..: 4 4 6 7 7 5 5 5 4 4 ...
##  $ Type           : chr  NA NA NA &quot;DS&quot; ...
##  $ Purpose        : chr  &quot;reactorfoampotentials&quot; &quot;reactorfoampotentials&quot; &quot;reactorfoampotentials&quot; &quot;foamnofoamcomparison&quot; ...
##  $ Experiment     : chr  &quot;SRX8715180&quot; &quot;SRX8715169&quot; &quot;SRX8715158&quot; &quot;SRX8715151&quot; ...
##  $ lat_lon        : chr  &quot;55.398077 N 10.415041 E&quot; &quot;56.175302 N 9.582588 E&quot; &quot;55.488235 N 8.430655 E&quot; &quot;56.425367 N 9.4527943 E&quot; ...
##  $ ReleaseDate    : chr  &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; ...
##  $ Sample.Name    : chr  &quot;16SAMP-17236&quot; &quot;16SAMP-17231&quot; &quot;16SAMP-17233&quot; &quot;MQ170915-14&quot; ...
##  $ Location       : Factor w/ 12 levels &quot;Avedoere&quot;,&quot;Damhusaaen&quot;,..: 4 11 5 12 12 12 12 2 11 11 ...
##  $ Observed       : num  161 354 247 265 389 454 406 304 278 371 ...
##  $ Chao1          : num  161 356 247 266 395 ...
##  $ se.chao1       : num  0 2.2689 0.0832 1.0272 3.914 ...
##  $ ACE            : num  161 357 247 266 395 ...
##  $ se.ACE         : num  4.6 9.28 7.43 7.72 9.83 ...
##  $ Shannon        : num  4.24 4.84 4.74 4.78 5.03 ...
##  $ Simpson        : num  0.966 0.973 0.982 0.983 0.986 ...
##  $ InvSimpson     : num  29.6 37.7 56.8 57.8 70.6 ...
##  $ Fisher         : num  27.4 72 46.1 50.2 81 ...
##  $ pielou_ev      : num  0.834 0.825 0.861 0.857 0.843 ...</code></pre>
</div>
<div id="colors" class="section level3 hasAnchor" number="6.2.4">
<h3><span class="header-section-number">6.2.4</span> Colors<a href="plotting-diversity.html#colors" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="plotting-diversity.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># You can check Hex codes with help of https://htmlcolorcodes.com/</span></span>
<span id="cb39-2"><a href="plotting-diversity.html#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="plotting-diversity.html#cb39-3" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#F9E79F&quot;</span>, <span class="st">&quot;#FDEBD0&quot;</span>, <span class="st">&quot;#F6DDCC&quot;</span>,  <span class="st">&quot;#E5E8E8&quot;</span>, <span class="st">&quot;#CCD1D1&quot;</span>, <span class="st">&quot;#AAB7B8&quot;</span>, </span>
<span id="cb39-4"><a href="plotting-diversity.html#cb39-4" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;#707B7C&quot;</span>, <span class="st">&quot;#566573&quot;</span>,<span class="st">&quot;#82E0AA&quot;</span>, <span class="st">&quot;#641E16&quot;</span>, <span class="st">&quot;#8E44AD&quot;</span>, <span class="st">&quot;#E74C3C&quot;</span>)</span>
<span id="cb39-5"><a href="plotting-diversity.html#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="plotting-diversity.html#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co"># you can also name each colour according to your treatment. </span></span>
<span id="cb39-7"><a href="plotting-diversity.html#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="co"># This will help ggplot to associated the same colour for each treatment across</span></span>
<span id="cb39-8"><a href="plotting-diversity.html#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="co"># different ggplots. For example, if we want to show different colour by location </span></span>
<span id="cb39-9"><a href="plotting-diversity.html#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="co"># in this dataset: </span></span>
<span id="cb39-10"><a href="plotting-diversity.html#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cols)  <span class="ot">&lt;-</span> <span class="fu">unique</span>((<span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps.flt)))<span class="sc">$</span>Location)</span>
<span id="cb39-11"><a href="plotting-diversity.html#cb39-11" aria-hidden="true" tabindex="-1"></a>cols</span></code></pre></div>
<pre><code>##   Ejby Moelle       Soeholt  Esbjerg Vest        Viborg    Damhusaaen 
##     &quot;#F9E79F&quot;     &quot;#FDEBD0&quot;     &quot;#F6DDCC&quot;     &quot;#E5E8E8&quot;     &quot;#CCD1D1&quot; 
##      Slagelse       Randers Mariagerfjord       Fornaes          Egaa 
##     &quot;#AAB7B8&quot;     &quot;#707B7C&quot;     &quot;#566573&quot;     &quot;#82E0AA&quot;     &quot;#641E16&quot; 
##      Avedoere     Hjoerring 
##     &quot;#8E44AD&quot;     &quot;#E74C3C&quot;</code></pre>
</div>
<div id="richness" class="section level3 hasAnchor" number="6.2.5">
<h3><span class="header-section-number">6.2.5</span> Richness<a href="plotting-diversity.html#richness" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="plotting-diversity.html#cb41-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps.flt)) <span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="plotting-diversity.html#cb41-2" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">ggboxplot</span>(., <span class="at">x =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb41-3"><a href="plotting-diversity.html#cb41-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">y =</span> <span class="st">&quot;Observed&quot;</span>, </span>
<span id="cb41-4"><a href="plotting-diversity.html#cb41-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb41-5"><a href="plotting-diversity.html#cb41-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ylab =</span> <span class="st">&quot;Richness&quot;</span>, </span>
<span id="cb41-6"><a href="plotting-diversity.html#cb41-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb41-7"><a href="plotting-diversity.html#cb41-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">palette =</span> cols) <span class="sc">+</span></span>
<span id="cb41-8"><a href="plotting-diversity.html#cb41-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb41-9"><a href="plotting-diversity.html#cb41-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>() ) <span class="sc">+</span></span>
<span id="cb41-10"><a href="plotting-diversity.html#cb41-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb41-11"><a href="plotting-diversity.html#cb41-11" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb41-12"><a href="plotting-diversity.html#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme(axis.title.x=element_blank(),  # Options</span></span>
<span id="cb41-13"><a href="plotting-diversity.html#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co">#        axis.text.x=element_blank(),</span></span>
<span id="cb41-14"><a href="plotting-diversity.html#cb41-14" aria-hidden="true" tabindex="-1"></a><span class="co">#        axis.ticks.x=element_blank()) +</span></span>
<span id="cb41-15"><a href="plotting-diversity.html#cb41-15" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;kruskal&quot;</span>,<span class="at">label.x =</span> <span class="dv">2</span>)</span>
<span id="cb41-16"><a href="plotting-diversity.html#cb41-16" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/richness-1.png" width="672" /></p>
</div>
<div id="shannon" class="section level3 hasAnchor" number="6.2.6">
<h3><span class="header-section-number">6.2.6</span> Shannon<a href="plotting-diversity.html#shannon" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="plotting-diversity.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shannon</span></span>
<span id="cb42-2"><a href="plotting-diversity.html#cb42-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps.flt)) <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="plotting-diversity.html#cb42-3" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">ggboxplot</span>(., <span class="at">x =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb42-4"><a href="plotting-diversity.html#cb42-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">y =</span> <span class="st">&quot;Shannon&quot;</span>, </span>
<span id="cb42-5"><a href="plotting-diversity.html#cb42-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb42-6"><a href="plotting-diversity.html#cb42-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ylab =</span> <span class="st">&quot;Shannon diversity (H)&quot;</span>, </span>
<span id="cb42-7"><a href="plotting-diversity.html#cb42-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb42-8"><a href="plotting-diversity.html#cb42-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">palette =</span> cols) <span class="sc">+</span></span>
<span id="cb42-9"><a href="plotting-diversity.html#cb42-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb42-10"><a href="plotting-diversity.html#cb42-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>() ) <span class="sc">+</span></span>
<span id="cb42-11"><a href="plotting-diversity.html#cb42-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb42-12"><a href="plotting-diversity.html#cb42-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb42-13"><a href="plotting-diversity.html#cb42-13" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme(axis.title.x=element_blank(),  # Options</span></span>
<span id="cb42-14"><a href="plotting-diversity.html#cb42-14" aria-hidden="true" tabindex="-1"></a><span class="co">#        axis.text.x=element_blank(),</span></span>
<span id="cb42-15"><a href="plotting-diversity.html#cb42-15" aria-hidden="true" tabindex="-1"></a><span class="co">#        axis.ticks.x=element_blank()) +</span></span>
<span id="cb42-16"><a href="plotting-diversity.html#cb42-16" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;kruskal&quot;</span>,<span class="at">label.x =</span> <span class="dv">2</span>)</span>
<span id="cb42-17"><a href="plotting-diversity.html#cb42-17" aria-hidden="true" tabindex="-1"></a>p2</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/shannon-1.png" width="672" /></p>
</div>
<div id="pielous-evenness" class="section level3 hasAnchor" number="6.2.7">
<h3><span class="header-section-number">6.2.7</span> Pielou’s evenness<a href="plotting-diversity.html#pielous-evenness" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="plotting-diversity.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pielou</span></span>
<span id="cb43-2"><a href="plotting-diversity.html#cb43-2" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps.flt)) <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="plotting-diversity.html#cb43-3" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">ggboxplot</span>(., <span class="at">x =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb43-4"><a href="plotting-diversity.html#cb43-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">y =</span> <span class="st">&quot;pielou_ev&quot;</span>, </span>
<span id="cb43-5"><a href="plotting-diversity.html#cb43-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb43-6"><a href="plotting-diversity.html#cb43-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ylab =</span> <span class="st">&quot;Pielou Evenness (J)&quot;</span>, </span>
<span id="cb43-7"><a href="plotting-diversity.html#cb43-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb43-8"><a href="plotting-diversity.html#cb43-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">palette =</span> cols) <span class="sc">+</span></span>
<span id="cb43-9"><a href="plotting-diversity.html#cb43-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb43-10"><a href="plotting-diversity.html#cb43-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>() ) <span class="sc">+</span></span>
<span id="cb43-11"><a href="plotting-diversity.html#cb43-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb43-12"><a href="plotting-diversity.html#cb43-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb43-13"><a href="plotting-diversity.html#cb43-13" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme(axis.title.x=element_blank(),  # Options</span></span>
<span id="cb43-14"><a href="plotting-diversity.html#cb43-14" aria-hidden="true" tabindex="-1"></a><span class="co">#        axis.text.x=element_blank(),</span></span>
<span id="cb43-15"><a href="plotting-diversity.html#cb43-15" aria-hidden="true" tabindex="-1"></a><span class="co">#        axis.ticks.x=element_blank()) +</span></span>
<span id="cb43-16"><a href="plotting-diversity.html#cb43-16" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;kruskal&quot;</span>,<span class="at">label.x =</span> <span class="dv">2</span>)</span>
<span id="cb43-17"><a href="plotting-diversity.html#cb43-17" aria-hidden="true" tabindex="-1"></a>p3</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/pielou-1.png" width="672" /></p>
</div>
<div id="combined-plotting" class="section level3 hasAnchor" number="6.2.8">
<h3><span class="header-section-number">6.2.8</span> Combined plotting<a href="plotting-diversity.html#combined-plotting" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="plotting-diversity.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co">#pdf(NULL)</span></span>
<span id="cb44-2"><a href="plotting-diversity.html#cb44-2" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">&lt;-</span> ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(p1, p2, p3, <span class="at">common.legend =</span> <span class="cn">TRUE</span>, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">legend =</span> <span class="st">&#39;top&#39;</span>)</span>
<span id="cb44-3"><a href="plotting-diversity.html#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="co">#x = dev.off()</span></span>
<span id="cb44-4"><a href="plotting-diversity.html#cb44-4" aria-hidden="true" tabindex="-1"></a>g1</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/plot-1.png" width="960" /></p>
<p><br />
</p>
</div>
<div id="nmds" class="section level3 hasAnchor" number="6.2.9">
<h3><span class="header-section-number">6.2.9</span> NMDS<a href="plotting-diversity.html#nmds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="plotting-diversity.html#cb45-1" aria-hidden="true" tabindex="-1"></a>physeqNMDS <span class="ot">&lt;-</span>  ps_rare  <span class="co"># We are using the rarefied abundances as explained in the intro </span></span>
<span id="cb45-2"><a href="plotting-diversity.html#cb45-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-3"><a href="plotting-diversity.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co">#ordination</span></span>
<span id="cb45-4"><a href="plotting-diversity.html#cb45-4" aria-hidden="true" tabindex="-1"></a>ordination <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">ordinate</span>(physeqNMDS  , <span class="st">&quot;NMDS&quot;</span>, <span class="at">distance =</span> <span class="st">&#39;bray&#39;</span>) <span class="co"># using Bray-Curtis dissimilarities here. </span></span>
<span id="cb45-5"><a href="plotting-diversity.html#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="plotting-diversity.html#cb45-6" aria-hidden="true" tabindex="-1"></a>label <span class="ot">&lt;-</span> <span class="fu">round</span>(ordination<span class="sc">$</span>stress,<span class="dv">3</span>) <span class="co"># stress label</span></span>
<span id="cb45-7"><a href="plotting-diversity.html#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting ordination</span></span>
<span id="cb45-8"><a href="plotting-diversity.html#cb45-8" aria-hidden="true" tabindex="-1"></a>NMDSplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_ordination</span>(physeqNMDS, </span>
<span id="cb45-9"><a href="plotting-diversity.html#cb45-9" aria-hidden="true" tabindex="-1"></a>                                      ordination , </span>
<span id="cb45-10"><a href="plotting-diversity.html#cb45-10" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">color =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb45-11"><a href="plotting-diversity.html#cb45-11" aria-hidden="true" tabindex="-1"></a>                                      <span class="co">#shape = &quot;Reactor&quot;,    </span></span>
<span id="cb45-12"><a href="plotting-diversity.html#cb45-12" aria-hidden="true" tabindex="-1"></a>                                      <span class="co"># option to add shapes as data symbols</span></span>
<span id="cb45-13"><a href="plotting-diversity.html#cb45-13" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">title =</span> <span class="st">&quot;NMDS&quot;</span>) <span class="sc">+</span></span>
<span id="cb45-14"><a href="plotting-diversity.html#cb45-14" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> pielou_ev), <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span>   <span class="co"># shape size in proportion to Pielou Evenness, also make transparent </span></span>
<span id="cb45-15"><a href="plotting-diversity.html#cb45-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">&#39;text&#39;</span>, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;Bray-Curtis, Stress = &quot;</span>, label), </span>
<span id="cb45-16"><a href="plotting-diversity.html#cb45-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">y =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.05</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">size =</span> <span class="dv">3</span>)  <span class="sc">+</span></span>
<span id="cb45-17"><a href="plotting-diversity.html#cb45-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cols)</span>
<span id="cb45-18"><a href="plotting-diversity.html#cb45-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-19"><a href="plotting-diversity.html#cb45-19" aria-hidden="true" tabindex="-1"></a>NMDSplot</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/nmds-1.png" width="576" /></p>
</div>
<div id="pca" class="section level3 hasAnchor" number="6.2.10">
<h3><span class="header-section-number">6.2.10</span> PCA<a href="plotting-diversity.html#pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="plotting-diversity.html#cb46-1" aria-hidden="true" tabindex="-1"></a>physeqPCA <span class="ot">&lt;-</span>  ps.flt  <span class="co"># just to keep it separate.</span></span>
<span id="cb46-2"><a href="plotting-diversity.html#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="plotting-diversity.html#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co"># normalisation and transform using the microbiome package</span></span>
<span id="cb46-4"><a href="plotting-diversity.html#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co"># I chose a centred log-ratio transform and a principle component analysis in a euclidean space.</span></span>
<span id="cb46-5"><a href="plotting-diversity.html#cb46-5" aria-hidden="true" tabindex="-1"></a>physeqPCA <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">transform</span>(physeqPCA, <span class="st">&quot;clr&quot;</span>)</span>
<span id="cb46-6"><a href="plotting-diversity.html#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="plotting-diversity.html#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="co">#ordination</span></span>
<span id="cb46-8"><a href="plotting-diversity.html#cb46-8" aria-hidden="true" tabindex="-1"></a>ordination <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">ordinate</span>(physeqPCA , <span class="st">&quot;RDA&quot;</span>) <span class="co"># useing the centred-log transformed abundances</span></span>
<span id="cb46-9"><a href="plotting-diversity.html#cb46-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-10"><a href="plotting-diversity.html#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting ordination</span></span>
<span id="cb46-11"><a href="plotting-diversity.html#cb46-11" aria-hidden="true" tabindex="-1"></a>PCAplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_ordination</span>(physeqPCA, </span>
<span id="cb46-12"><a href="plotting-diversity.html#cb46-12" aria-hidden="true" tabindex="-1"></a>                                      ordination , </span>
<span id="cb46-13"><a href="plotting-diversity.html#cb46-13" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">color =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb46-14"><a href="plotting-diversity.html#cb46-14" aria-hidden="true" tabindex="-1"></a>                                      <span class="co">#shape = &quot;Reactor&quot;, </span></span>
<span id="cb46-15"><a href="plotting-diversity.html#cb46-15" aria-hidden="true" tabindex="-1"></a>                                     <span class="co"># option to add shapes as data symbols</span></span>
<span id="cb46-16"><a href="plotting-diversity.html#cb46-16" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">title =</span> <span class="st">&quot;Principle Component Analysis&quot;</span>) <span class="sc">+</span></span>
<span id="cb46-17"><a href="plotting-diversity.html#cb46-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> pielou_ev)) <span class="sc">+</span>    <span class="co"># # shape size in proportion to Pielou Evenness</span></span>
<span id="cb46-18"><a href="plotting-diversity.html#cb46-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">&#39;text&#39;</span>, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;Aitchison distances&quot;</span>), </span>
<span id="cb46-19"><a href="plotting-diversity.html#cb46-19" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">y =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.05</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">size =</span> <span class="dv">3</span>)  <span class="sc">+</span> </span>
<span id="cb46-20"><a href="plotting-diversity.html#cb46-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cols)</span>
<span id="cb46-21"><a href="plotting-diversity.html#cb46-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-22"><a href="plotting-diversity.html#cb46-22" aria-hidden="true" tabindex="-1"></a>PCAplot</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/PCA-1.png" width="576" /></p>
<p><br />
</p>
</div>
<div id="rda" class="section level3 hasAnchor" number="6.2.11">
<h3><span class="header-section-number">6.2.11</span> RDA<a href="plotting-diversity.html#rda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="plotting-diversity.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># PERCENT FILTERVALUE FOR ALL RDAs</span></span>
<span id="cb47-2"><a href="plotting-diversity.html#cb47-2" aria-hidden="true" tabindex="-1"></a>pcnt <span class="ot">&lt;-</span> <span class="fl">0.25</span></span>
<span id="cb47-3"><a href="plotting-diversity.html#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># FORMULA FOR ALL RDAs</span></span>
<span id="cb47-4"><a href="plotting-diversity.html#cb47-4" aria-hidden="true" tabindex="-1"></a>form <span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span> Location)</span>
<span id="cb47-5"><a href="plotting-diversity.html#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="co"># labels for arrows</span></span>
<span id="cb47-6"><a href="plotting-diversity.html#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="co"># pos for arrows</span></span>
<span id="cb47-7"><a href="plotting-diversity.html#cb47-7" aria-hidden="true" tabindex="-1"></a>xarw <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb47-8"><a href="plotting-diversity.html#cb47-8" aria-hidden="true" tabindex="-1"></a>yarw <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb47-9"><a href="plotting-diversity.html#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="co"># arrow length map</span></span>
<span id="cb47-10"><a href="plotting-diversity.html#cb47-10" aria-hidden="true" tabindex="-1"></a>larw <span class="ot">&lt;-</span> <span class="fl">1.5</span></span>
<span id="cb47-11"><a href="plotting-diversity.html#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="co"># arrow length map for chromosol rda</span></span>
<span id="cb47-12"><a href="plotting-diversity.html#cb47-12" aria-hidden="true" tabindex="-1"></a>larwch <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb47-13"><a href="plotting-diversity.html#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="co"># fixed width of x axis</span></span>
<span id="cb47-14"><a href="plotting-diversity.html#cb47-14" aria-hidden="true" tabindex="-1"></a>xaxis <span class="ot">=</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">6.75</span>, <span class="fl">6.5</span>)</span>
<span id="cb47-15"><a href="plotting-diversity.html#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="co"># position of annotation</span></span>
<span id="cb47-16"><a href="plotting-diversity.html#cb47-16" aria-hidden="true" tabindex="-1"></a>textposx <span class="ot">=</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">4.4</span>)</span>
<span id="cb47-17"><a href="plotting-diversity.html#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="co"># arrow color</span></span>
<span id="cb47-18"><a href="plotting-diversity.html#cb47-18" aria-hidden="true" tabindex="-1"></a>arrowcolor <span class="ot">&lt;-</span> <span class="st">&quot;grey13&quot;</span></span>
<span id="cb47-19"><a href="plotting-diversity.html#cb47-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-20"><a href="plotting-diversity.html#cb47-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-21"><a href="plotting-diversity.html#cb47-21" aria-hidden="true" tabindex="-1"></a>physeqPCA <span class="ot">&lt;-</span> ps.flt <span class="co"># excluding Heat, H2O2, reference and blank</span></span>
<span id="cb47-22"><a href="plotting-diversity.html#cb47-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-23"><a href="plotting-diversity.html#cb47-23" aria-hidden="true" tabindex="-1"></a><span class="co"># normalisation and transform</span></span>
<span id="cb47-24"><a href="plotting-diversity.html#cb47-24" aria-hidden="true" tabindex="-1"></a>physeqPCA <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">transform</span>(physeqPCA, <span class="st">&quot;clr&quot;</span>)</span>
<span id="cb47-25"><a href="plotting-diversity.html#cb47-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-26"><a href="plotting-diversity.html#cb47-26" aria-hidden="true" tabindex="-1"></a><span class="co">#ordination</span></span>
<span id="cb47-27"><a href="plotting-diversity.html#cb47-27" aria-hidden="true" tabindex="-1"></a>ordination <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">ordinate</span>(physeqPCA, <span class="st">&quot;RDA&quot;</span>, <span class="st">&quot;bray&quot;</span>, <span class="at">formula =</span> form )</span>
<span id="cb47-28"><a href="plotting-diversity.html#cb47-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-29"><a href="plotting-diversity.html#cb47-29" aria-hidden="true" tabindex="-1"></a>RDAplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_ordination</span>(physeqPCA, ordination , </span>
<span id="cb47-30"><a href="plotting-diversity.html#cb47-30" aria-hidden="true" tabindex="-1"></a>                                <span class="at">color =</span> <span class="st">&quot;Location&quot;</span>)  <span class="sc">+</span></span>
<span id="cb47-31"><a href="plotting-diversity.html#cb47-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb47-32"><a href="plotting-diversity.html#cb47-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span>    <span class="co"># shape size reflective of Pielou  </span></span>
<span id="cb47-33"><a href="plotting-diversity.html#cb47-33" aria-hidden="true" tabindex="-1"></a>     <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">&#39;text&#39;</span>, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;Aitchison distances&quot;</span>), </span>
<span id="cb47-34"><a href="plotting-diversity.html#cb47-34" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">y =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.05</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb47-35"><a href="plotting-diversity.html#cb47-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> pielou_ev)) <span class="sc">+</span>        <span class="co"># shape size in proportion to Pielou Evenness</span></span>
<span id="cb47-36"><a href="plotting-diversity.html#cb47-36" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_colour_gradient(low = colourlow, high = colourhigh) +</span></span>
<span id="cb47-37"><a href="plotting-diversity.html#cb47-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;RDA Abundances (clr) ~ Location&quot;</span>) <span class="sc">+</span></span>
<span id="cb47-38"><a href="plotting-diversity.html#cb47-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cols)</span>
<span id="cb47-39"><a href="plotting-diversity.html#cb47-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-40"><a href="plotting-diversity.html#cb47-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-41"><a href="plotting-diversity.html#cb47-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Now add the environmental variables as arrows to either of these p1 or p2</span></span>
<span id="cb47-42"><a href="plotting-diversity.html#cb47-42" aria-hidden="true" tabindex="-1"></a>arrowmat <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">scores</span>(ordination, <span class="at">display =</span> <span class="st">&quot;bp&quot;</span>)</span>
<span id="cb47-43"><a href="plotting-diversity.html#cb47-43" aria-hidden="true" tabindex="-1"></a>arrowmat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(arrowmat)</span>
<span id="cb47-44"><a href="plotting-diversity.html#cb47-44" aria-hidden="true" tabindex="-1"></a><span class="co">#rownames(arrowmat) &lt;- arrowlabel</span></span>
<span id="cb47-45"><a href="plotting-diversity.html#cb47-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Add labels, make a data.frame</span></span>
<span id="cb47-46"><a href="plotting-diversity.html#cb47-46" aria-hidden="true" tabindex="-1"></a>arrowdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">labels =</span> <span class="fu">rownames</span>(arrowmat), arrowmat)</span>
<span id="cb47-47"><a href="plotting-diversity.html#cb47-47" aria-hidden="true" tabindex="-1"></a>arrowdf<span class="sc">$</span>labels <span class="ot">&lt;-</span> arrowdf<span class="sc">$</span>labels <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;Location&quot;</span>, <span class="st">&quot;&quot;</span>, .) <span class="co"># remove the string &quot;Location&quot; from the labels. </span></span>
<span id="cb47-48"><a href="plotting-diversity.html#cb47-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the arrow aesthetic mapping</span></span>
<span id="cb47-49"><a href="plotting-diversity.html#cb47-49" aria-hidden="true" tabindex="-1"></a>arrow_map <span class="ot">&lt;-</span> <span class="fu">aes</span>(<span class="at">xend =</span> larw <span class="sc">*</span> RDA1,</span>
<span id="cb47-50"><a href="plotting-diversity.html#cb47-50" aria-hidden="true" tabindex="-1"></a>                 <span class="at">yend =</span> larw <span class="sc">*</span> RDA2,</span>
<span id="cb47-51"><a href="plotting-diversity.html#cb47-51" aria-hidden="true" tabindex="-1"></a>                 <span class="at">x =</span> <span class="dv">0</span>,</span>
<span id="cb47-52"><a href="plotting-diversity.html#cb47-52" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> <span class="dv">0</span>,</span>
<span id="cb47-53"><a href="plotting-diversity.html#cb47-53" aria-hidden="true" tabindex="-1"></a>                 <span class="at">shape =</span> <span class="cn">NULL</span>,</span>
<span id="cb47-54"><a href="plotting-diversity.html#cb47-54" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="cn">NULL</span>,</span>
<span id="cb47-55"><a href="plotting-diversity.html#cb47-55" aria-hidden="true" tabindex="-1"></a>                 <span class="at">label =</span> labels)</span>
<span id="cb47-56"><a href="plotting-diversity.html#cb47-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-57"><a href="plotting-diversity.html#cb47-57" aria-hidden="true" tabindex="-1"></a>label_map <span class="ot">&lt;-</span> <span class="fu">aes</span>(<span class="at">x =</span> xarw <span class="sc">*</span> RDA1,</span>
<span id="cb47-58"><a href="plotting-diversity.html#cb47-58" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> yarw <span class="sc">*</span> RDA2,</span>
<span id="cb47-59"><a href="plotting-diversity.html#cb47-59" aria-hidden="true" tabindex="-1"></a>                 <span class="at">shape =</span> <span class="cn">NULL</span>,</span>
<span id="cb47-60"><a href="plotting-diversity.html#cb47-60" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="cn">NULL</span>,</span>
<span id="cb47-61"><a href="plotting-diversity.html#cb47-61" aria-hidden="true" tabindex="-1"></a>                 <span class="at">label =</span> labels)</span>
<span id="cb47-62"><a href="plotting-diversity.html#cb47-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-63"><a href="plotting-diversity.html#cb47-63" aria-hidden="true" tabindex="-1"></a>arrowhead <span class="ot">=</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.02</span>, <span class="st">&quot;npc&quot;</span>))</span>
<span id="cb47-64"><a href="plotting-diversity.html#cb47-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-65"><a href="plotting-diversity.html#cb47-65" aria-hidden="true" tabindex="-1"></a>RDAplot <span class="ot">&lt;-</span>RDAplot <span class="sc">+</span></span>
<span id="cb47-66"><a href="plotting-diversity.html#cb47-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(</span>
<span id="cb47-67"><a href="plotting-diversity.html#cb47-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">mapping =</span> arrow_map,</span>
<span id="cb47-68"><a href="plotting-diversity.html#cb47-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> .<span class="dv">5</span>,</span>
<span id="cb47-69"><a href="plotting-diversity.html#cb47-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> arrowdf,</span>
<span id="cb47-70"><a href="plotting-diversity.html#cb47-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> arrowcolor,</span>
<span id="cb47-71"><a href="plotting-diversity.html#cb47-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrow =</span> arrowhead)  <span class="sc">+</span></span>
<span id="cb47-72"><a href="plotting-diversity.html#cb47-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb47-73"><a href="plotting-diversity.html#cb47-73" aria-hidden="true" tabindex="-1"></a>   <span class="at">mapping =</span> label_map,</span>
<span id="cb47-74"><a href="plotting-diversity.html#cb47-74" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb47-75"><a href="plotting-diversity.html#cb47-75" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> arrowdf,</span>
<span id="cb47-76"><a href="plotting-diversity.html#cb47-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">show.legend =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-77"><a href="plotting-diversity.html#cb47-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-78"><a href="plotting-diversity.html#cb47-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-79"><a href="plotting-diversity.html#cb47-79" aria-hidden="true" tabindex="-1"></a>RDAplot</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/rda-1.png" width="576" /></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="plotting-diversity.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># save figure for publication</span></span>
<span id="cb48-2"><a href="plotting-diversity.html#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave(&quot;RDA.png&quot;, height=6, width=7.5, units=&#39;in&#39;, dpi=600)</span></span>
<span id="cb48-3"><a href="plotting-diversity.html#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co">#knitr::include_graphics(&quot;./RDA.png&quot;)</span></span></code></pre></div>
<p><br />
</p>
</div>
<div id="unweighted-unifrac" class="section level3 hasAnchor" number="6.2.12">
<h3><span class="header-section-number">6.2.12</span> Unweighted uniFrac<a href="plotting-diversity.html#unweighted-unifrac" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>UniFrac distances are derived from phylogenetic distances between ASVs, which means that a phylogenetic tree needs to be included into the phyloseq object.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="plotting-diversity.html#cb49-1" aria-hidden="true" tabindex="-1"></a>physeqPcOA <span class="ot">&lt;-</span>  ps.flt  <span class="co"># just to keep it separate.</span></span>
<span id="cb49-2"><a href="plotting-diversity.html#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="plotting-diversity.html#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="co">#ordination</span></span>
<span id="cb49-4"><a href="plotting-diversity.html#cb49-4" aria-hidden="true" tabindex="-1"></a>ordination <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">ordinate</span>(physeqPcOA  , </span>
<span id="cb49-5"><a href="plotting-diversity.html#cb49-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">&quot;PCoA&quot;</span>, </span>
<span id="cb49-6"><a href="plotting-diversity.html#cb49-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">distance =</span> <span class="st">&#39;unifrac&#39;</span>) <span class="co"># using just presence and absence (unweighted) dissimilarities here. </span></span>
<span id="cb49-7"><a href="plotting-diversity.html#cb49-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-8"><a href="plotting-diversity.html#cb49-8" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting ordination</span></span>
<span id="cb49-9"><a href="plotting-diversity.html#cb49-9" aria-hidden="true" tabindex="-1"></a>PCOAplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_ordination</span>(physeqPcOA, </span>
<span id="cb49-10"><a href="plotting-diversity.html#cb49-10" aria-hidden="true" tabindex="-1"></a>                                      ordination , </span>
<span id="cb49-11"><a href="plotting-diversity.html#cb49-11" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">color =</span> <span class="st">&quot;Location&quot;</span>, </span>
<span id="cb49-12"><a href="plotting-diversity.html#cb49-12" aria-hidden="true" tabindex="-1"></a>                                      <span class="co">#shape = &quot;Reactor&quot;,    </span></span>
<span id="cb49-13"><a href="plotting-diversity.html#cb49-13" aria-hidden="true" tabindex="-1"></a>                                      <span class="co"># option to add shapes to data symbols</span></span>
<span id="cb49-14"><a href="plotting-diversity.html#cb49-14" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">title =</span> <span class="st">&quot;Principle Coordinate Analysis&quot;</span>) <span class="sc">+</span></span>
<span id="cb49-15"><a href="plotting-diversity.html#cb49-15" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> pielou_ev)) <span class="sc">+</span>    <span class="co"># shape size reflective of Pielou  </span></span>
<span id="cb49-16"><a href="plotting-diversity.html#cb49-16" aria-hidden="true" tabindex="-1"></a>     <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">&#39;text&#39;</span>, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;unweighted UniFrac&quot;</span>), </span>
<span id="cb49-17"><a href="plotting-diversity.html#cb49-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">y =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.05</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">size =</span> <span class="dv">3</span>)  <span class="sc">+</span></span>
<span id="cb49-18"><a href="plotting-diversity.html#cb49-18" aria-hidden="true" tabindex="-1"></a>     <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cols)</span>
<span id="cb49-19"><a href="plotting-diversity.html#cb49-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-20"><a href="plotting-diversity.html#cb49-20" aria-hidden="true" tabindex="-1"></a>PCOAplot</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/unifrac-1.png" width="576" /></p>
</div>
<div id="combined-plotting-1" class="section level3 hasAnchor" number="6.2.13">
<h3><span class="header-section-number">6.2.13</span> Combined plotting<a href="plotting-diversity.html#combined-plotting-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="plotting-diversity.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">#pdf(NULL)</span></span>
<span id="cb50-2"><a href="plotting-diversity.html#cb50-2" aria-hidden="true" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(NMDSplot, PCAplot, RDAplot, PCOAplot,</span>
<span id="cb50-3"><a href="plotting-diversity.html#cb50-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">common.legend =</span> <span class="cn">TRUE</span>, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">legend =</span> <span class="st">&#39;top&#39;</span>)</span>
<span id="cb50-4"><a href="plotting-diversity.html#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co">#x = dev.off()</span></span>
<span id="cb50-5"><a href="plotting-diversity.html#cb50-5" aria-hidden="true" tabindex="-1"></a>g2</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/plotbeta-1.png" width="960" /></p>
<p><br />
</p>
</div>
<div id="save-plots-to-png-and-pdf" class="section level3 hasAnchor" number="6.2.14">
<h3><span class="header-section-number">6.2.14</span> Save plots to png and pdf<a href="plotting-diversity.html#save-plots-to-png-and-pdf" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="plotting-diversity.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># either as png or pdf</span></span>
<span id="cb51-2"><a href="plotting-diversity.html#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;alphadiversity.png&quot;</span>, <span class="at">plot =</span> (g1), <span class="at">height=</span><span class="dv">4</span>, <span class="at">width=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">&#39;in&#39;</span>, <span class="at">dpi=</span><span class="dv">600</span>)</span>
<span id="cb51-3"><a href="plotting-diversity.html#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;alphadiversity.pdf&quot;</span>, <span class="at">plot =</span> (g1), <span class="at">height=</span><span class="dv">4</span>, <span class="at">width=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">&#39;in&#39;</span>, <span class="at">dpi=</span><span class="dv">600</span>)</span>
<span id="cb51-4"><a href="plotting-diversity.html#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="plotting-diversity.html#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;betadiversity.png&quot;</span>, <span class="at">plot =</span> (g2), <span class="at">height=</span><span class="dv">10</span>, <span class="at">width=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">&#39;in&#39;</span>, <span class="at">dpi=</span><span class="dv">600</span>)</span>
<span id="cb51-6"><a href="plotting-diversity.html#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;betadiversity.pdf&quot;</span>, <span class="at">plot =</span> (g2), <span class="at">height=</span><span class="dv">10</span>, <span class="at">width=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">&#39;in&#39;</span>, <span class="at">dpi=</span><span class="dv">600</span>)</span></code></pre></div>
<p><strong>Note:</strong>
This code is an amalgamation from various sources. Apart from putting it together into a pipeline I do not take credit for it.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Qiime2R.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/cjvanlissa/gitbook-demo/edit/master/06_diversity.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["gitbook-demo.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
