<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Plotting taxonomy | Workflows for processing microbial amplicon sequences</title>
  <meta name="description" content="This GitBook contains bioinformatic workflows from raw microbial amplicon-sequence reads generated with a Miseq instrument to pre-processing reads with packages such as Qiime2, followed by applying various packages within the R Environment." />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Plotting taxonomy | Workflows for processing microbial amplicon sequences" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This GitBook contains bioinformatic workflows from raw microbial amplicon-sequence reads generated with a Miseq instrument to pre-processing reads with packages such as Qiime2, followed by applying various packages within the R Environment." />
  <meta name="github-repo" content="chrismitbiz/ABlab-workflows" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Plotting taxonomy | Workflows for processing microbial amplicon sequences" />
  
  <meta name="twitter:description" content="This GitBook contains bioinformatic workflows from raw microbial amplicon-sequence reads generated with a Miseq instrument to pre-processing reads with packages such as Qiime2, followed by applying various packages within the R Environment." />
  

<meta name="author" content="Christian Krohn, PhD, RMIT University" />


<meta name="date" content="2022-07-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="plottingdiversity.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A GitBook for Teaching</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this GitBook</a></li>
<li class="chapter" data-level="2" data-path="gettingstarted.html"><a href="gettingstarted.html"><i class="fa fa-check"></i><b>2</b> Getting started</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gettingstarted.html"><a href="gettingstarted.html#general-requirements"><i class="fa fa-check"></i><b>2.1</b> General requirements</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="gettingstarted.html"><a href="gettingstarted.html#access-to-computational-resources"><i class="fa fa-check"></i><b>2.1.1</b> Access to computational resources</a></li>
<li class="chapter" data-level="2.1.2" data-path="gettingstarted.html"><a href="gettingstarted.html#file-storage"><i class="fa fa-check"></i><b>2.1.2</b> File storage</a></li>
<li class="chapter" data-level="2.1.3" data-path="gettingstarted.html"><a href="gettingstarted.html#command-line"><i class="fa fa-check"></i><b>2.1.3</b> Command Line</a></li>
<li class="chapter" data-level="2.1.4" data-path="gettingstarted.html"><a href="gettingstarted.html#data-storage-browser"><i class="fa fa-check"></i><b>2.1.4</b> Data storage browser</a></li>
<li class="chapter" data-level="2.1.5" data-path="gettingstarted.html"><a href="gettingstarted.html#r-and-r-studio"><i class="fa fa-check"></i><b>2.1.5</b> R and R studio</a></li>
<li class="chapter" data-level="2.1.6" data-path="gettingstarted.html"><a href="gettingstarted.html#environment-managers"><i class="fa fa-check"></i><b>2.1.6</b> Environment managers</a></li>
<li class="chapter" data-level="2.1.7" data-path="gettingstarted.html"><a href="gettingstarted.html#explore-the-help-functions"><i class="fa fa-check"></i><b>2.1.7</b> Explore the help functions</a></li>
<li class="chapter" data-level="2.1.8" data-path="gettingstarted.html"><a href="gettingstarted.html#github-account"><i class="fa fa-check"></i><b>2.1.8</b> GitHub account</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="gettingstarted.html"><a href="gettingstarted.html#a-note-on-rarefying-normalising"><i class="fa fa-check"></i><b>2.2</b> A note on rarefying (normalising)</a></li>
<li class="chapter" data-level="2.3" data-path="gettingstarted.html"><a href="gettingstarted.html#workflows-from-other-lab-groups"><i class="fa fa-check"></i><b>2.3</b> Workflows from other lab groups</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html"><i class="fa fa-check"></i><b>3</b> Miseq library preps</a>
<ul>
<li class="chapter" data-level="3.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#overview"><i class="fa fa-check"></i><b>3.1.1</b> Overview</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#workflow"><i class="fa fa-check"></i><b>3.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-extraction"><i class="fa fa-check"></i><b>3.2.1</b> DNA extraction</a></li>
<li class="chapter" data-level="3.2.2" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-quality-and-concentration"><i class="fa fa-check"></i><b>3.2.2</b> DNA quality and concentration</a></li>
<li class="chapter" data-level="3.2.3" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-normalisation"><i class="fa fa-check"></i><b>3.2.3</b> DNA normalisation</a></li>
<li class="chapter" data-level="3.2.4" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#first-and-second-pcr---amplicon-pcr-and-indexing-pcr"><i class="fa fa-check"></i><b>3.2.4</b> First and second PCR - Amplicon PCR and Indexing PCR</a></li>
<li class="chapter" data-level="3.2.5" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#normalisation-of-indexed-amplicons"><i class="fa fa-check"></i><b>3.2.5</b> Normalisation of indexed amplicons</a></li>
<li class="chapter" data-level="3.2.6" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#pooling"><i class="fa fa-check"></i><b>3.2.6</b> Pooling</a></li>
<li class="chapter" data-level="3.2.7" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#denaturing-of-pool-and-loading-of-the-miseq"><i class="fa fa-check"></i><b>3.2.7</b> Denaturing of pool and loading of the Miseq</a></li>
<li class="chapter" data-level="3.2.8" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#consumables"><i class="fa fa-check"></i><b>3.2.8</b> Consumables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html"><i class="fa fa-check"></i><b>4</b> From BaseSpace to Qiime2 and DADA2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#prerequisites-and-required-files"><i class="fa fa-check"></i><b>4.1.1</b> Prerequisites and required files</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#workflow-1"><i class="fa fa-check"></i><b>4.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#download-fastq-files"><i class="fa fa-check"></i><b>4.2.1</b> Download FASTQ files</a></li>
<li class="chapter" data-level="4.2.2" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#paired-end-manifest-import-step-1"><i class="fa fa-check"></i><b>4.2.2</b> Paired end manifest import (Step 1)</a></li>
<li class="chapter" data-level="4.2.3" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#cutadapt-step-2"><i class="fa fa-check"></i><b>4.2.3</b> Cutadapt (Step 2)</a></li>
<li class="chapter" data-level="4.2.4" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#read-quality-assessment"><i class="fa fa-check"></i><b>4.2.4</b> Read quality assessment</a></li>
<li class="chapter" data-level="4.2.5" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#denoise-paired-end-sequences-with-dada2-step-3"><i class="fa fa-check"></i><b>4.2.5</b> Denoise paired end sequences with dada2 (Step 3)</a></li>
<li class="chapter" data-level="4.2.6" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#taxonomic-classifier-and-assignment-step-4"><i class="fa fa-check"></i><b>4.2.6</b> Taxonomic classifier and assignment (Step 4)</a></li>
<li class="chapter" data-level="4.2.7" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#build-phylogenetic-tree-step-5"><i class="fa fa-check"></i><b>4.2.7</b> Build phylogenetic tree (Step 5)</a></li>
<li class="chapter" data-level="4.2.8" data-path="from-basespace-to-qiime2-and-dada2.html"><a href="from-basespace-to-qiime2-and-dada2.html#all-steps-combined"><i class="fa fa-check"></i><b>4.2.8</b> All steps combined</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Qiime2R.html"><a href="Qiime2R.html"><i class="fa fa-check"></i><b>5</b> From Qiime2 into R - Initial diagnostics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Qiime2R.html"><a href="Qiime2R.html#introduction-2"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="Qiime2R.html"><a href="Qiime2R.html#workflow-2"><i class="fa fa-check"></i><b>5.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Qiime2R.html"><a href="Qiime2R.html#packages"><i class="fa fa-check"></i><b>5.2.1</b> Packages</a></li>
<li class="chapter" data-level="5.2.2" data-path="Qiime2R.html"><a href="Qiime2R.html#import-qiime-files-and-create-a-phyloseq-object"><i class="fa fa-check"></i><b>5.2.2</b> Import qiime-files and create a phyloseq object</a></li>
<li class="chapter" data-level="5.2.3" data-path="Qiime2R.html"><a href="Qiime2R.html#save-the-ps-object-as-an-.rds-file-into-your-working-directory"><i class="fa fa-check"></i><b>5.2.3</b> Save the <code>ps</code> object as an <code>.rds</code> file into your working directory</a></li>
<li class="chapter" data-level="5.2.4" data-path="Qiime2R.html"><a href="Qiime2R.html#initial-filtering"><i class="fa fa-check"></i><b>5.2.4</b> Initial filtering</a></li>
<li class="chapter" data-level="5.2.5" data-path="Qiime2R.html"><a href="Qiime2R.html#proportion-of-asvs-identified"><i class="fa fa-check"></i><b>5.2.5</b> Proportion of ASVs identified</a></li>
<li class="chapter" data-level="5.2.6" data-path="Qiime2R.html"><a href="Qiime2R.html#rarefaction-curve"><i class="fa fa-check"></i><b>5.2.6</b> Rarefaction curve</a></li>
<li class="chapter" data-level="5.2.7" data-path="Qiime2R.html"><a href="Qiime2R.html#prevalence-table"><i class="fa fa-check"></i><b>5.2.7</b> Prevalence table</a></li>
<li class="chapter" data-level="5.2.8" data-path="Qiime2R.html"><a href="Qiime2R.html#export-asv-and-taxa-tables-to-excel"><i class="fa fa-check"></i><b>5.2.8</b> Export ASV and taxa tables to Excel</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="plottingdiversity.html"><a href="plottingdiversity.html"><i class="fa fa-check"></i><b>6</b> Plotting diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="plottingdiversity.html"><a href="plottingdiversity.html#introduction-3"><i class="fa fa-check"></i><b>6.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="plottingdiversity.html"><a href="plottingdiversity.html#required-files"><i class="fa fa-check"></i><b>6.1.1</b> Required files</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="plottingdiversity.html"><a href="plottingdiversity.html#workflow-3"><i class="fa fa-check"></i><b>6.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="plottingdiversity.html"><a href="plottingdiversity.html#packages-1"><i class="fa fa-check"></i><b>6.2.1</b> Packages</a></li>
<li class="chapter" data-level="6.2.2" data-path="plottingdiversity.html"><a href="plottingdiversity.html#load-phyloseq-object"><i class="fa fa-check"></i><b>6.2.2</b> Load phyloseq object</a></li>
<li class="chapter" data-level="6.2.3" data-path="plottingdiversity.html"><a href="plottingdiversity.html#pre-filter-rarefy-and-calculate-diversity"><i class="fa fa-check"></i><b>6.2.3</b> Pre-filter, rarefy and calculate diversity</a></li>
<li class="chapter" data-level="6.2.4" data-path="plottingdiversity.html"><a href="plottingdiversity.html#colors"><i class="fa fa-check"></i><b>6.2.4</b> Colors</a></li>
<li class="chapter" data-level="6.2.5" data-path="plottingdiversity.html"><a href="plottingdiversity.html#richness"><i class="fa fa-check"></i><b>6.2.5</b> Richness</a></li>
<li class="chapter" data-level="6.2.6" data-path="plottingdiversity.html"><a href="plottingdiversity.html#shannon"><i class="fa fa-check"></i><b>6.2.6</b> Shannon</a></li>
<li class="chapter" data-level="6.2.7" data-path="plottingdiversity.html"><a href="plottingdiversity.html#pielous-evenness"><i class="fa fa-check"></i><b>6.2.7</b> Pielou’s evenness</a></li>
<li class="chapter" data-level="6.2.8" data-path="plottingdiversity.html"><a href="plottingdiversity.html#combined-plotting"><i class="fa fa-check"></i><b>6.2.8</b> Combined plotting</a></li>
<li class="chapter" data-level="6.2.9" data-path="plottingdiversity.html"><a href="plottingdiversity.html#nmds"><i class="fa fa-check"></i><b>6.2.9</b> NMDS</a></li>
<li class="chapter" data-level="6.2.10" data-path="plottingdiversity.html"><a href="plottingdiversity.html#pca"><i class="fa fa-check"></i><b>6.2.10</b> PCA</a></li>
<li class="chapter" data-level="6.2.11" data-path="plottingdiversity.html"><a href="plottingdiversity.html#rda"><i class="fa fa-check"></i><b>6.2.11</b> RDA</a></li>
<li class="chapter" data-level="6.2.12" data-path="plottingdiversity.html"><a href="plottingdiversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>6.2.12</b> Unweighted uniFrac</a></li>
<li class="chapter" data-level="6.2.13" data-path="plottingdiversity.html"><a href="plottingdiversity.html#combined-plotting-1"><i class="fa fa-check"></i><b>6.2.13</b> Combined plotting</a></li>
<li class="chapter" data-level="6.2.14" data-path="plottingdiversity.html"><a href="plottingdiversity.html#save-plots-to-png-and-pdf"><i class="fa fa-check"></i><b>6.2.14</b> Save plots to png and pdf</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plottingtaxa.html"><a href="plottingtaxa.html"><i class="fa fa-check"></i><b>7</b> Plotting taxonomy</a>
<ul>
<li class="chapter" data-level="7.1" data-path="plottingtaxa.html"><a href="plottingtaxa.html#introduction-4"><i class="fa fa-check"></i><b>7.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="plottingtaxa.html"><a href="plottingtaxa.html#required-files-or-objects"><i class="fa fa-check"></i><b>7.1.1</b> Required files or objects</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="plottingtaxa.html"><a href="plottingtaxa.html#workflow-4"><i class="fa fa-check"></i><b>7.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="plottingtaxa.html"><a href="plottingtaxa.html#packages-2"><i class="fa fa-check"></i><b>7.2.1</b> Packages</a></li>
<li class="chapter" data-level="7.2.2" data-path="plottingtaxa.html"><a href="plottingtaxa.html#load-phyloseq-object-1"><i class="fa fa-check"></i><b>7.2.2</b> Load phyloseq object</a></li>
<li class="chapter" data-level="7.2.3" data-path="plottingtaxa.html"><a href="plottingtaxa.html#check-metadata-contained-in-the-phyloseq-object."><i class="fa fa-check"></i><b>7.2.3</b> Check metadata contained in the phyloseq object.</a></li>
<li class="chapter" data-level="7.2.4" data-path="plottingtaxa.html"><a href="plottingtaxa.html#colors-1"><i class="fa fa-check"></i><b>7.2.4</b> Colors</a></li>
<li class="chapter" data-level="7.2.5" data-path="plottingtaxa.html"><a href="plottingtaxa.html#pre-filtering"><i class="fa fa-check"></i><b>7.2.5</b> Pre-filtering</a></li>
<li class="chapter" data-level="7.2.6" data-path="plottingtaxa.html"><a href="plottingtaxa.html#analysis-and-plotting"><i class="fa fa-check"></i><b>7.2.6</b> Analysis and plotting</a></li>
<li class="chapter" data-level="7.2.7" data-path="plottingtaxa.html"><a href="plottingtaxa.html#prevalance-table"><i class="fa fa-check"></i><b>7.2.7</b> Prevalance table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workflows for processing microbial amplicon sequences</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="plottingtaxa" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Plotting taxonomy<a href="plottingtaxa.html#plottingtaxa" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="img/Phylumplot.png" /></p>
<div id="introduction-4" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Introduction<a href="plottingtaxa.html#introduction-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this chapter you will learn how to plot a custom taxa barplot to compare relative abundances between different sets of samples or treatments.</p>
<div id="required-files-or-objects" class="section level3 hasAnchor" number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> Required files or objects<a href="plottingtaxa.html#required-files-or-objects" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>A <code>phyloseq</code> object; either read in from a pre-saved <code>.rds</code> file or created as described in chapter <a href="Qiime2R.html#Qiime2R">5</a> under “Import qiime-files and create a phyloseq object”.<br />
</li>
<li>If you dont have your own data you can download the pre-saved <code>.rds</code> object from this Chapter and follow the below steps: <a href="./ps_ProjectX_2022July">ps_ProjectX_2022July</a>.</li>
</ul>
</div>
</div>
<div id="workflow-4" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Workflow<a href="plottingtaxa.html#workflow-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="packages-2" class="section level3 hasAnchor" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Packages<a href="plottingtaxa.html#packages-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="plottingtaxa.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb58-2"><a href="plottingtaxa.html#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)       <span class="co"># a handy helper package for ggplots</span></span>
<span id="cb58-3"><a href="plottingtaxa.html#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb58-4"><a href="plottingtaxa.html#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())  <span class="co"># setting the theme for ggplots</span></span>
<span id="cb58-5"><a href="plottingtaxa.html#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span></code></pre></div>
</div>
<div id="load-phyloseq-object-1" class="section level3 hasAnchor" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Load phyloseq object<a href="plottingtaxa.html#load-phyloseq-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Do this if you saved a phyloseq object in .rds format before. Otherwise create a phyloseq object as described in chapter <a href="Qiime2R.html#Qiime2R">5</a> under “Import qiime-files and create a phyloseq object”.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="plottingtaxa.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reading in a previously saved phyloseq object</span></span>
<span id="cb59-2"><a href="plottingtaxa.html#cb59-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&#39;ps_ProjectX_2022July&#39;</span>)</span>
<span id="cb59-3"><a href="plottingtaxa.html#cb59-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-4"><a href="plottingtaxa.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="co">#ps (not run) to get an overview of number of taxa and samples contained in the phyloseq object</span></span>
<span id="cb59-5"><a href="plottingtaxa.html#cb59-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-6"><a href="plottingtaxa.html#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="co"># output</span></span>
<span id="cb59-7"><a href="plottingtaxa.html#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="co"># phyloseq-class experiment-level object</span></span>
<span id="cb59-8"><a href="plottingtaxa.html#cb59-8" aria-hidden="true" tabindex="-1"></a><span class="co"># otu_table()   OTU Table:         [ 4218 taxa and 51 samples ]</span></span>
<span id="cb59-9"><a href="plottingtaxa.html#cb59-9" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_data() Sample Data:       [ 51 samples by 15 sample variables ]</span></span>
<span id="cb59-10"><a href="plottingtaxa.html#cb59-10" aria-hidden="true" tabindex="-1"></a><span class="co"># tax_table()   Taxonomy Table:    [ 4218 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb59-11"><a href="plottingtaxa.html#cb59-11" aria-hidden="true" tabindex="-1"></a><span class="co"># phy_tree()    Phylogenetic Tree: [ 4218 tips and 4217 internal nodes ]</span></span></code></pre></div>
</div>
<div id="check-metadata-contained-in-the-phyloseq-object." class="section level3 hasAnchor" number="7.2.3">
<h3><span class="header-section-number">7.2.3</span> Check metadata contained in the phyloseq object.<a href="plottingtaxa.html#check-metadata-contained-in-the-phyloseq-object." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now lets check which columns in the metadata contain factors. Factors enable you to determine the order of categorical data, such as treatments, in the ggplots. Learn more about factors here: <a href="https://swcarpentry.github.io/r-novice-inflammation/12-supp-factors/index.html" class="uri">https://swcarpentry.github.io/r-novice-inflammation/12-supp-factors/index.html</a>.</p>
<p>In this case, the columns <code>Reactor</code> and <code>Location</code> contain factors.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="plottingtaxa.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the &#39;structure&#39; (str) command to get an overview of your data columns</span></span>
<span id="cb60-2"><a href="plottingtaxa.html#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Learn what are characters, integers or factors, </span></span>
<span id="cb60-3"><a href="plottingtaxa.html#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(<span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ps)))</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    51 obs. of  15 variables:
##  $ Assay.Type     : chr  &quot;AMPLICON&quot; &quot;AMPLICON&quot; &quot;AMPLICON&quot; &quot;AMPLICON&quot; ...
##  $ Bases          : int  13254234 24196788 15758554 20048406 37634632 43555904 35107436 13888140 23172184 35065898 ...
##  $ BioProject     : chr  &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; &quot;PRJNA645373&quot; ...
##  $ Bytes          : int  9235913 16404070 10758456 13009806 24207581 27963491 22585685 9388862 15125835 22520074 ...
##  $ Organism       : chr  &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; &quot;anaerobic digester metagenome&quot; ...
##  $ collection_date: chr  &quot;30/11/16&quot; &quot;9/3/16&quot; &quot;17/3/16&quot; &quot;5/12/16&quot; ...
##  $ loc            : chr  &quot;Ejby Moelle&quot; &quot;Soeholt&quot; &quot;Esbjerg Vest&quot; &quot;Viborg&quot; ...
##  $ Reactor        : Factor w/ 10 levels &quot;1A&quot;,&quot;1B&quot;,&quot;2B&quot;,..: 4 4 6 7 7 5 5 5 4 4 ...
##  $ Type           : chr  NA NA NA &quot;DS&quot; ...
##  $ Purpose        : chr  &quot;reactorfoampotentials&quot; &quot;reactorfoampotentials&quot; &quot;reactorfoampotentials&quot; &quot;foamnofoamcomparison&quot; ...
##  $ Experiment     : chr  &quot;SRX8715180&quot; &quot;SRX8715169&quot; &quot;SRX8715158&quot; &quot;SRX8715151&quot; ...
##  $ lat_lon        : chr  &quot;55.398077 N 10.415041 E&quot; &quot;56.175302 N 9.582588 E&quot; &quot;55.488235 N 8.430655 E&quot; &quot;56.425367 N 9.4527943 E&quot; ...
##  $ ReleaseDate    : chr  &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; &quot;2021-01-01T00:00:00Z&quot; ...
##  $ Sample.Name    : chr  &quot;16SAMP-17236&quot; &quot;16SAMP-17231&quot; &quot;16SAMP-17233&quot; &quot;MQ170915-14&quot; ...
##  $ Location       : Factor w/ 12 levels &quot;Avedoere&quot;,&quot;Damhusaaen&quot;,..: 4 11 5 12 12 12 12 2 11 11 ...</code></pre>
</div>
<div id="colors-1" class="section level3 hasAnchor" number="7.2.4">
<h3><span class="header-section-number">7.2.4</span> Colors<a href="plottingtaxa.html#colors-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before you create ggplots to compare taxonomic compositions between samples and treatments it is helpful to create a named color vector that will apply to all your plots and ensures that each taxonomic name is a specific color across all your plots.</p>
<p>There are different packages that help to create a names color vector. Here we use the package <code>colorspace</code>.</p>
<p>A handy color cheat sheet is available here: <a href="https://www.nceas.ucsb.edu/sites/default/files/2020-04/colorPaletteCheatsheet.pdf" class="uri">https://www.nceas.ucsb.edu/sites/default/files/2020-04/colorPaletteCheatsheet.pdf</a></p>
<p>First create a vector of all possible taxonomic names that could be included in your ggplot. For example, if you plan to plot on phylum-level then all phyla names are extracted from the phyloseq object and then those names are given a color code. The colors are defined as hex color codes.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="plottingtaxa.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># You can check Hex codes with help of https://htmlcolorcodes.com/</span></span>
<span id="cb62-2"><a href="plottingtaxa.html#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="co"># https://www.nceas.ucsb.edu/sites/default/files/2020-04/colorPaletteCheatsheet.pdf</span></span>
<span id="cb62-3"><a href="plottingtaxa.html#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="plottingtaxa.html#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co"># For all possible phyla  names</span></span>
<span id="cb62-5"><a href="plottingtaxa.html#cb62-5" aria-hidden="true" tabindex="-1"></a>phylum_data <span class="ot">&lt;-</span>  microbiome<span class="sc">::</span><span class="fu">aggregate_taxa</span>(ps, <span class="st">&quot;Phylum&quot;</span>) <span class="co"># aggregate data to phylum level </span></span>
<span id="cb62-6"><a href="plottingtaxa.html#cb62-6" aria-hidden="true" tabindex="-1"></a><span class="co"># the pacakage &#39;microbiome&#39; has some handy helper functions such as `aggregate`.</span></span>
<span id="cb62-7"><a href="plottingtaxa.html#cb62-7" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(phylum_data<span class="sc">@</span>tax_table) <span class="co"># extract a vector of phylum names using `rownames`</span></span>
<span id="cb62-8"><a href="plottingtaxa.html#cb62-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-9"><a href="plottingtaxa.html#cb62-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Package colorspace</span></span>
<span id="cb62-10"><a href="plottingtaxa.html#cb62-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(colorspace)</span>
<span id="cb62-11"><a href="plottingtaxa.html#cb62-11" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> colorspace<span class="sc">::</span><span class="fu">rainbow_hcl</span>(<span class="fu">length</span>(names)) <span class="co"># this creates x number of hex codes. x = `length(names)`. </span></span>
<span id="cb62-12"><a href="plottingtaxa.html#cb62-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  in this case there are 48 phyla, hence we get 48 hex codes. </span></span>
<span id="cb62-13"><a href="plottingtaxa.html#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cols) <span class="ot">&lt;-</span> names  <span class="co"># name each hex code with phyla names. </span></span>
<span id="cb62-14"><a href="plottingtaxa.html#cb62-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-15"><a href="plottingtaxa.html#cb62-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Option to use the package  viridis</span></span>
<span id="cb62-16"><a href="plottingtaxa.html#cb62-16" aria-hidden="true" tabindex="-1"></a><span class="co">#library(viridis)</span></span>
<span id="cb62-17"><a href="plottingtaxa.html#cb62-17" aria-hidden="true" tabindex="-1"></a><span class="co"># cols &lt;- viridis::turbo(length(names), direction = -1)</span></span>
<span id="cb62-18"><a href="plottingtaxa.html#cb62-18" aria-hidden="true" tabindex="-1"></a><span class="co"># names(cols) &lt;- names</span></span>
<span id="cb62-19"><a href="plottingtaxa.html#cb62-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-20"><a href="plottingtaxa.html#cb62-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Option to use the package RColorBrewer</span></span>
<span id="cb62-21"><a href="plottingtaxa.html#cb62-21" aria-hidden="true" tabindex="-1"></a><span class="co"># library(RColorBrewer)</span></span>
<span id="cb62-22"><a href="plottingtaxa.html#cb62-22" aria-hidden="true" tabindex="-1"></a><span class="co"># display.brewer.all(length(names))  # check colour themes available in the RColorBrewer package</span></span>
<span id="cb62-23"><a href="plottingtaxa.html#cb62-23" aria-hidden="true" tabindex="-1"></a><span class="co"># cols &lt;- brewer.pal(length(names), &#39;RdYlBu&#39;)</span></span>
<span id="cb62-24"><a href="plottingtaxa.html#cb62-24" aria-hidden="true" tabindex="-1"></a><span class="co"># cols &lt;- rainbow_hcl(length(names))</span></span>
<span id="cb62-25"><a href="plottingtaxa.html#cb62-25" aria-hidden="true" tabindex="-1"></a><span class="co"># names(cols) &lt;- names</span></span>
<span id="cb62-26"><a href="plottingtaxa.html#cb62-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-27"><a href="plottingtaxa.html#cb62-27" aria-hidden="true" tabindex="-1"></a>cols <span class="co"># check content of the color vector</span></span></code></pre></div>
<pre><code>##                  Acetothermia               Acidobacteriota 
##                     &quot;#E495A5&quot;                     &quot;#E3979D&quot; 
##              Actinobacteriota                Armatimonadota 
##                     &quot;#E19995&quot;                     &quot;#DE9B8D&quot; 
##                  Bacteroidota              Bdellovibrionota 
##                     &quot;#DB9D85&quot;                     &quot;#D79F7E&quot; 
##            Caldatribacteriota                 Caldisericota 
##                     &quot;#D2A277&quot;                     &quot;#CDA471&quot; 
##              Campilobacterota                   Chloroflexi 
##                     &quot;#C7A76C&quot;                     &quot;#C1A968&quot; 
##                      CK-2C2-2                Cloacimonadota 
##                     &quot;#BAAC65&quot;                     &quot;#B3AE64&quot; 
##          Coprothermobacterota                 Cyanobacteria 
##                     &quot;#ABB065&quot;                     &quot;#A2B367&quot; 
##                  Deinococcota                  Dependentiae 
##                     &quot;#99B56B&quot;                     &quot;#90B66F&quot; 
##              Desulfobacterota               Elusimicrobiota 
##                     &quot;#86B875&quot;                     &quot;#7CBA7C&quot; 
##                       FCPU426             Fermentibacterota 
##                     &quot;#72BB83&quot;                     &quot;#67BC8A&quot; 
##                Fibrobacterota                    Firmicutes 
##                     &quot;#5CBD92&quot;                     &quot;#51BE9A&quot; 
##                Fusobacteriota               Gemmatimonadota 
##                     &quot;#47BEA2&quot;                     &quot;#3FBEA9&quot; 
##               Hydrogenedentes                  Hydrothermae 
##                     &quot;#39BEB1&quot;                     &quot;#37BDB8&quot; 
##              Latescibacterota              Margulisbacteria 
##                     &quot;#3BBCBF&quot;                     &quot;#42BBC6&quot; 
## Marinimicrobia_(SAR406_clade)                        MBNT15 
##                     &quot;#4CB9CC&quot;                     &quot;#57B8D1&quot; 
##             Methylomirabilota                   Myxococcota 
##                     &quot;#64B5D6&quot;                     &quot;#70B3DA&quot; 
##                  Nitrospirota               Patescibacteria 
##                     &quot;#7DB0DD&quot;                     &quot;#8AADE0&quot; 
##               Planctomycetota                Proteobacteria 
##                     &quot;#96AAE1&quot;                     &quot;#A1A7E2&quot; 
##  SAR324_clade(Marine_group_B)                 Spirochaetota 
##                     &quot;#ACA4E2&quot;                     &quot;#B5A1E1&quot; 
##                   Sumerlaeota                       Sva0485 
##                     &quot;#BE9EDF&quot;                     &quot;#C69BDC&quot; 
##                  Synergistota                  Thermotogota 
##                     &quot;#CD99D8&quot;                     &quot;#D497D4&quot; 
##             Verrucomicrobiota                         WPS-2 
##                     &quot;#D995CF&quot;                     &quot;#DD94C9&quot; 
##                           WS1                           WS4 
##                     &quot;#E093C3&quot;                     &quot;#E293BC&quot; 
##                  Zixibacteria                       Unknown 
##                     &quot;#E493B4&quot;                     &quot;#E494AD&quot;</code></pre>
</div>
<div id="pre-filtering" class="section level3 hasAnchor" number="7.2.5">
<h3><span class="header-section-number">7.2.5</span> Pre-filtering<a href="plottingtaxa.html#pre-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Some arbitrary quality filtering of individual ASVs. For example, here we choose to remove ASVs that have very few reads (i.e. less than 5 counts). Use the <code>prune_taxa()</code> function from phyloseq to filter taxa or the <code>prune_samples()</code> function to filter selected samples.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="plottingtaxa.html#cb64-1" aria-hidden="true" tabindex="-1"></a>ps_flt  <span class="ot">&lt;-</span>  <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps) <span class="sc">&gt;=</span> <span class="dv">5</span>, ps) <span class="co">#minimum reads per feature  </span></span>
<span id="cb64-2"><a href="plottingtaxa.html#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ps_flt </span></span>
<span id="cb64-3"><a href="plottingtaxa.html#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="co"># phyloseq-class experiment-level object</span></span>
<span id="cb64-4"><a href="plottingtaxa.html#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="co"># otu_table()   OTU Table:         [ 3911 taxa and 51 samples ]</span></span>
<span id="cb64-5"><a href="plottingtaxa.html#cb64-5" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_data() Sample Data:       [ 51 samples by 15 sample variables ]</span></span>
<span id="cb64-6"><a href="plottingtaxa.html#cb64-6" aria-hidden="true" tabindex="-1"></a><span class="co"># tax_table()   Taxonomy Table:    [ 3911 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb64-7"><a href="plottingtaxa.html#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="co"># phy_tree()    Phylogenetic Tree: [ 3911 tips and 3910 internal nodes ]</span></span>
<span id="cb64-8"><a href="plottingtaxa.html#cb64-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-9"><a href="plottingtaxa.html#cb64-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Use `prune_samples()` If you need to filter data to a subset of samples.</span></span>
<span id="cb64-10"><a href="plottingtaxa.html#cb64-10" aria-hidden="true" tabindex="-1"></a><span class="co"># E.g. choose samples of reactor 1A only</span></span>
<span id="cb64-11"><a href="plottingtaxa.html#cb64-11" aria-hidden="true" tabindex="-1"></a>ps_flt_1A  <span class="ot">&lt;-</span>  <span class="fu">prune_samples</span>(<span class="fu">sample_data</span>(ps_flt)<span class="sc">$</span>Reactor <span class="sc">==</span> <span class="st">&quot;1A&quot;</span>, ps_flt) </span>
<span id="cb64-12"><a href="plottingtaxa.html#cb64-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-13"><a href="plottingtaxa.html#cb64-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Use `prune_taxa` to remove any ASVs/taxa that are now zero </span></span>
<span id="cb64-14"><a href="plottingtaxa.html#cb64-14" aria-hidden="true" tabindex="-1"></a><span class="co"># (i.e. are not present in reactor 1A). </span></span>
<span id="cb64-15"><a href="plottingtaxa.html#cb64-15" aria-hidden="true" tabindex="-1"></a><span class="co"># In other words we keep only those ASVs/Taxa that are present in the filtered samples. </span></span>
<span id="cb64-16"><a href="plottingtaxa.html#cb64-16" aria-hidden="true" tabindex="-1"></a>ps_flt_1A  <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps_flt_1A) <span class="sc">&gt;</span> <span class="dv">0</span>, ps_flt_1A)</span>
<span id="cb64-17"><a href="plottingtaxa.html#cb64-17" aria-hidden="true" tabindex="-1"></a><span class="co">#phyloseq-class experiment-level object</span></span>
<span id="cb64-18"><a href="plottingtaxa.html#cb64-18" aria-hidden="true" tabindex="-1"></a><span class="co">#otu_table()   OTU Table:         [ 420 taxa and 2 samples ]</span></span>
<span id="cb64-19"><a href="plottingtaxa.html#cb64-19" aria-hidden="true" tabindex="-1"></a><span class="co">#sample_data() Sample Data:       [ 2 samples by 15 sample variables ]</span></span>
<span id="cb64-20"><a href="plottingtaxa.html#cb64-20" aria-hidden="true" tabindex="-1"></a><span class="co">#tax_table()   Taxonomy Table:    [ 420 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb64-21"><a href="plottingtaxa.html#cb64-21" aria-hidden="true" tabindex="-1"></a><span class="co">#phy_tree()    Phylogenetic Tree: [ 420 tips and 419 internal nodes ]</span></span></code></pre></div>
</div>
<div id="analysis-and-plotting" class="section level3 hasAnchor" number="7.2.6">
<h3><span class="header-section-number">7.2.6</span> Analysis and plotting<a href="plottingtaxa.html#analysis-and-plotting" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create stacked barplot comparing relative abundances of the normalised phyloseq object.</p>
<p>First abundances are aggregated to phylum level and then transformed into relative abundances. This is followed by the creation of a long-form abundance table that includes metadata with help of the very handy <code>psmelt</code> function from the phyloseq package. The function <code>psmelt</code> takes a phyloseq object and extracts abundances and related metadata into a dataframe for plotting.</p>
<p>Finally, the abundances of this long abundance table are summarised into the factors that you want to compare. In this example, abundances of different Locations are compared, hence the column <code>Location</code> will be summarised and plotted.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="plottingtaxa.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Phylum plot  </span></span>
<span id="cb65-2"><a href="plottingtaxa.html#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="co"># First, create a long abundance data table  </span></span>
<span id="cb65-3"><a href="plottingtaxa.html#cb65-3" aria-hidden="true" tabindex="-1"></a>taxa_data <span class="ot">&lt;-</span>  microbiome<span class="sc">::</span><span class="fu">aggregate_taxa</span>(ps_flt, <span class="st">&quot;Phylum&quot;</span>) <span class="sc">%&gt;%</span>   </span>
<span id="cb65-4"><a href="plottingtaxa.html#cb65-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># aggregate abundances to phylum-level</span></span>
<span id="cb65-5"><a href="plottingtaxa.html#cb65-5" aria-hidden="true" tabindex="-1"></a>  microbiome<span class="sc">::</span><span class="fu">transform</span>(., <span class="st">&quot;compositional&quot;</span>) <span class="sc">%&gt;%</span>  <span class="co"># transform abundance to relative abundances</span></span>
<span id="cb65-6"><a href="plottingtaxa.html#cb65-6" aria-hidden="true" tabindex="-1"></a>  phyloseq<span class="sc">::</span><span class="fu">psmelt</span>(.)  <span class="co"># extract long abundance table with metadata for plotting. </span></span>
<span id="cb65-7"><a href="plottingtaxa.html#cb65-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-8"><a href="plottingtaxa.html#cb65-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: The `.` replaces the phyloseq object. </span></span>
<span id="cb65-9"><a href="plottingtaxa.html#cb65-9" aria-hidden="true" tabindex="-1"></a><span class="co"># This is because the functions are piped into a sequence by using the `%&gt;%` operator. </span></span>
<span id="cb65-10"><a href="plottingtaxa.html#cb65-10" aria-hidden="true" tabindex="-1"></a><span class="co"># The `%&gt;%` operator is a Tidyverse function and is very helpful in putting </span></span>
<span id="cb65-11"><a href="plottingtaxa.html#cb65-11" aria-hidden="true" tabindex="-1"></a><span class="co"># various different operations and changes to the data into one sequence. </span></span>
<span id="cb65-12"><a href="plottingtaxa.html#cb65-12" aria-hidden="true" tabindex="-1"></a><span class="co"># We could have put the below operations into the same above sequence too.</span></span>
<span id="cb65-13"><a href="plottingtaxa.html#cb65-13" aria-hidden="true" tabindex="-1"></a><span class="co"># There is no difference in the outcome.</span></span>
<span id="cb65-14"><a href="plottingtaxa.html#cb65-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-15"><a href="plottingtaxa.html#cb65-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarise abundances by columns of your choice containing factors.</span></span>
<span id="cb65-16"><a href="plottingtaxa.html#cb65-16" aria-hidden="true" tabindex="-1"></a><span class="co"># E.g. compute the mean abundances based on the columns `Location`, `Reactor` and `Phylum.   </span></span>
<span id="cb65-17"><a href="plottingtaxa.html#cb65-17" aria-hidden="true" tabindex="-1"></a>taxa_data <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span></span>
<span id="cb65-18"><a href="plottingtaxa.html#cb65-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(Location, Reactor, Phylum) <span class="sc">%&gt;%</span>  <span class="co"># group by Location and Phylum to summarise data</span></span>
<span id="cb65-19"><a href="plottingtaxa.html#cb65-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">Meanabu =</span> <span class="fu">mean</span>(Abundance)) <span class="sc">%&gt;%</span>  <span class="co"># Create a column called `Meanabu`</span></span>
<span id="cb65-20"><a href="plottingtaxa.html#cb65-20" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(Meanabu <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span>                  <span class="co"># filter out rows that have 0 abundance</span></span>
<span id="cb65-21"><a href="plottingtaxa.html#cb65-21" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(Meanabu)) <span class="co"># sort the abundances </span></span>
<span id="cb65-22"><a href="plottingtaxa.html#cb65-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-23"><a href="plottingtaxa.html#cb65-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Now, create a vector with phylum names that will determine the order of </span></span>
<span id="cb65-24"><a href="plottingtaxa.html#cb65-24" aria-hidden="true" tabindex="-1"></a><span class="co"># phylum names in the stacked barcharts.</span></span>
<span id="cb65-25"><a href="plottingtaxa.html#cb65-25" aria-hidden="true" tabindex="-1"></a>order <span class="ot">&lt;-</span> taxa_data<span class="sc">$</span>Phylum <span class="sc">%&gt;%</span> unique</span>
<span id="cb65-26"><a href="plottingtaxa.html#cb65-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-27"><a href="plottingtaxa.html#cb65-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Phylum names need to be factors for plotting. </span></span>
<span id="cb65-28"><a href="plottingtaxa.html#cb65-28" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the final dataframe for plotting</span></span>
<span id="cb65-29"><a href="plottingtaxa.html#cb65-29" aria-hidden="true" tabindex="-1"></a>taxa_data<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">factor</span>(taxa_data<span class="sc">$</span>Phylum, <span class="at">levels =</span> order) <span class="co"># </span></span>
<span id="cb65-30"><a href="plottingtaxa.html#cb65-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Check out the first few rows</span></span>
<span id="cb65-31"><a href="plottingtaxa.html#cb65-31" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(taxa_data)</span>
<span id="cb65-32"><a href="plottingtaxa.html#cb65-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-33"><a href="plottingtaxa.html#cb65-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Finally its time to create a ggplot. </span></span>
<span id="cb65-34"><a href="plottingtaxa.html#cb65-34" aria-hidden="true" tabindex="-1"></a>taxaplot <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span></span>
<span id="cb65-35"><a href="plottingtaxa.html#cb65-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggbarplot</span>( <span class="at">x=</span> <span class="st">&quot;Reactor&quot;</span>, </span>
<span id="cb65-36"><a href="plottingtaxa.html#cb65-36" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">&quot;Meanabu&quot;</span>, </span>
<span id="cb65-37"><a href="plottingtaxa.html#cb65-37" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb65-38"><a href="plottingtaxa.html#cb65-38" aria-hidden="true" tabindex="-1"></a>               <span class="at">position =</span> <span class="fu">position_stack</span>(),</span>
<span id="cb65-39"><a href="plottingtaxa.html#cb65-39" aria-hidden="true" tabindex="-1"></a>               <span class="at">legend =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb65-40"><a href="plottingtaxa.html#cb65-40" aria-hidden="true" tabindex="-1"></a>               <span class="at">ylab =</span> <span class="st">&quot;Relative Abundances&quot;</span>) <span class="sc">+</span></span>
<span id="cb65-41"><a href="plottingtaxa.html#cb65-41" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Tell ggplot to use the `cols` vector created earlier</span></span>
<span id="cb65-42"><a href="plottingtaxa.html#cb65-42" aria-hidden="true" tabindex="-1"></a>            <span class="co"># only the first 20 colors are shown in legend, hence the `order[1:20]`</span></span>
<span id="cb65-43"><a href="plottingtaxa.html#cb65-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> cols, </span>
<span id="cb65-44"><a href="plottingtaxa.html#cb65-44" aria-hidden="true" tabindex="-1"></a>                      <span class="at">limits =</span> order[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]) <span class="sc">+</span>   </span>
<span id="cb65-45"><a href="plottingtaxa.html#cb65-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(Location),     <span class="co"># create facets by factors in column `Location`.</span></span>
<span id="cb65-46"><a href="plottingtaxa.html#cb65-46" aria-hidden="true" tabindex="-1"></a>               <span class="at">scales =</span> <span class="st">&quot;free_x&quot;</span>,</span>
<span id="cb65-47"><a href="plottingtaxa.html#cb65-47" aria-hidden="true" tabindex="-1"></a>               <span class="at">space =</span> <span class="st">&quot;free_x&quot;</span>) <span class="sc">+</span></span>
<span id="cb65-48"><a href="plottingtaxa.html#cb65-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">byrow =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span> </span>
<span id="cb65-49"><a href="plottingtaxa.html#cb65-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rotate_x_text</span>(<span class="at">angle =</span> <span class="dv">45</span>) <span class="sc">+</span>           <span class="co"># rotate the x-axis labels</span></span>
<span id="cb65-50"><a href="plottingtaxa.html#cb65-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span>                          <span class="co"># makes ggplots look better</span></span>
<span id="cb65-51"><a href="plottingtaxa.html#cb65-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),   <span class="co"># remove the background grid</span></span>
<span id="cb65-52"><a href="plottingtaxa.html#cb65-52" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb65-53"><a href="plottingtaxa.html#cb65-53" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) </span>
<span id="cb65-54"><a href="plottingtaxa.html#cb65-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-55"><a href="plottingtaxa.html#cb65-55" aria-hidden="true" tabindex="-1"></a>taxaplot</span></code></pre></div>
<p><img src="gitbook-demo_files/figure-html/Phylumplots-1.png" width="1248" /></p>
</div>
<div id="prevalance-table" class="section level3 hasAnchor" number="7.2.7">
<h3><span class="header-section-number">7.2.7</span> Prevalance table<a href="plottingtaxa.html#prevalance-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A prevalence table may assist in getting an overview of the abundances. The mean prevalence is the average number of times that the taxon appears across all samples.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="plottingtaxa.html#cb66-1" aria-hidden="true" tabindex="-1"></a>prevalancedf <span class="ot">=</span> <span class="fu">apply</span>(<span class="at">X =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_flt),</span>
<span id="cb66-2"><a href="plottingtaxa.html#cb66-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">MARGIN =</span> <span class="dv">1</span>,</span>
<span id="cb66-3"><a href="plottingtaxa.html#cb66-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">FUN =</span> <span class="cf">function</span>(x){<span class="fu">sum</span>(x <span class="sc">&gt;</span> <span class="dv">0</span>)})</span>
<span id="cb66-4"><a href="plottingtaxa.html#cb66-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-5"><a href="plottingtaxa.html#cb66-5" aria-hidden="true" tabindex="-1"></a>prevalancedf <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Prevalence =</span> prevalancedf,</span>
<span id="cb66-6"><a href="plottingtaxa.html#cb66-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">TotalAbundance =</span> phyloseq<span class="sc">::</span><span class="fu">taxa_sums</span>(ps_flt),</span>
<span id="cb66-7"><a href="plottingtaxa.html#cb66-7" aria-hidden="true" tabindex="-1"></a>                      phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(ps_flt))</span>
<span id="cb66-8"><a href="plottingtaxa.html#cb66-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-9"><a href="plottingtaxa.html#cb66-9" aria-hidden="true" tabindex="-1"></a>prevalancedf <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">ddply</span>(prevalancedf, <span class="st">&quot;Phylum&quot;</span>, <span class="cf">function</span>(df1){</span>
<span id="cb66-10"><a href="plottingtaxa.html#cb66-10" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">data.frame</span>(<span class="at">mean_prevalence=</span><span class="fu">mean</span>(df1<span class="sc">$</span>Prevalence), </span>
<span id="cb66-11"><a href="plottingtaxa.html#cb66-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">total_abundance=</span><span class="fu">sum</span>(df1<span class="sc">$</span>TotalAbundance, <span class="at">na.rm =</span> T), </span>
<span id="cb66-12"><a href="plottingtaxa.html#cb66-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb66-13"><a href="plottingtaxa.html#cb66-13" aria-hidden="true" tabindex="-1"></a>                    })</span>
<span id="cb66-14"><a href="plottingtaxa.html#cb66-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-15"><a href="plottingtaxa.html#cb66-15" aria-hidden="true" tabindex="-1"></a><span class="co"># print table from highest to lowest abundance </span></span>
<span id="cb66-16"><a href="plottingtaxa.html#cb66-16" aria-hidden="true" tabindex="-1"></a>prevalancedf <span class="sc">%&gt;%</span> </span>
<span id="cb66-17"><a href="plottingtaxa.html#cb66-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(total_abundance))  </span></code></pre></div>
<pre><code>##                           Phylum mean_prevalence total_abundance
## 1                    Chloroflexi        5.124069          228197
## 2                   Bacteroidota        4.688679          201921
## 3                     Firmicutes        4.453435          197683
## 4               Actinobacteriota        5.231343          165740
## 5                 Proteobacteria        4.249147          161756
## 6               Desulfobacterota        5.425743           57146
## 7                   Synergistota        6.595506           51225
## 8                 Cloacimonadota        6.500000           45294
## 9                Acidobacteriota        3.953488           35304
## 10                  Thermotogota       10.190476           33489
## 11                 Spirochaetota        4.578431           25015
## 12             Verrucomicrobiota        3.382022           16598
## 13             Fermentibacterota       35.000000           14812
## 14               Patescibacteria        3.669173           14399
## 15            Caldatribacteriota        5.500000            8932
## 16               Planctomycetota        2.573171            5455
## 17                Armatimonadota        2.933333            4443
## 18 Marinimicrobia_(SAR406_clade)       20.000000            3818
## 19                          &lt;NA&gt;        4.052632            3511
## 20               Hydrogenedentes        3.950000            3411
## 21                  Nitrospirota        8.750000            2853
## 22              Campilobacterota        4.187500            2710
## 23  SAR324_clade(Marine_group_B)        4.000000            2616
## 24                           WS1        3.625000            2559
## 25                Fibrobacterota        2.826087            1938
## 26                   Myxococcota        3.176471            1353
## 27              Bdellovibrionota        3.600000            1111
## 28                 Caldisericota        3.818182            1067
## 29                         WPS-2        2.666667             897
## 30                   Sumerlaeota        3.888889             683
## 31                 Cyanobacteria        9.000000             647
## 32                Fusobacteriota        2.888889             565
## 33          Coprothermobacterota        2.800000             511
## 34               Gemmatimonadota        2.875000             300
## 35               Elusimicrobiota        2.833333             225
## 36                      CK-2C2-2        4.000000             162
## 37             Methylomirabilota        4.000000             159
## 38              Latescibacterota        2.333333              86
## 39                  Dependentiae        2.500000              78
## 40                  Zixibacteria        5.000000              74
## 41                       Sva0485        2.333333              66
## 42                  Acetothermia        2.000000              59
## 43                           WS4        1.500000              39
## 44                  Hydrothermae        1.000000              26
## 45                  Deinococcota        1.000000              11</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="plottingdiversity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/cjvanlissa/gitbook-demo/edit/master/07_taxonomic_visualisations.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["gitbook-demo.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
