<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Some fundamentals by Dr Jen Wood | Beginner’s guide to Miseq sequencing and processing of short-read microbial (16S) amplicon sequences</title>
  <meta name="description" content="This GitBook contains library preparation and bioinformatic workflows from DNA extraction, to processing of fastQ files generated with a Miseq instrument, to visualisation of microbial amplicon-sequences with various packages within Qiime2 and R Studio." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Some fundamentals by Dr Jen Wood | Beginner’s guide to Miseq sequencing and processing of short-read microbial (16S) amplicon sequences" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This GitBook contains library preparation and bioinformatic workflows from DNA extraction, to processing of fastQ files generated with a Miseq instrument, to visualisation of microbial amplicon-sequences with various packages within Qiime2 and R Studio." />
  <meta name="github-repo" content="chrismitbiz/ABlab-workflows" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Some fundamentals by Dr Jen Wood | Beginner’s guide to Miseq sequencing and processing of short-read microbial (16S) amplicon sequences" />
  
  <meta name="twitter:description" content="This GitBook contains library preparation and bioinformatic workflows from DNA extraction, to processing of fastQ files generated with a Miseq instrument, to visualisation of microbial amplicon-sequences with various packages within Qiime2 and R Studio." />
  

<meta name="author" content="Christian Krohn, PhD, RMIT University" />


<meta name="date" content="2024-06-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="gettingstarted.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A GitBook for Teaching</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this GitBook</a></li>
<li class="chapter" data-level="2" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html"><i class="fa fa-check"></i><b>2</b> Some fundamentals by Dr Jen Wood</a>
<ul>
<li class="chapter" data-level="2.1" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#alpha-diversity"><i class="fa fa-check"></i><b>2.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#richness"><i class="fa fa-check"></i><b>2.1.1</b> Richness</a></li>
<li class="chapter" data-level="2.1.2" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#evenness"><i class="fa fa-check"></i><b>2.1.2</b> Evenness</a></li>
<li class="chapter" data-level="2.1.3" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#exercise"><i class="fa fa-check"></i><b>2.1.3</b> Exercise</a></li>
<li class="chapter" data-level="2.1.4" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#shannon-weiner-diversity"><i class="fa fa-check"></i><b>2.1.4</b> Shannon-Weiner Diversity</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#beta-diversity"><i class="fa fa-check"></i><b>2.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#species-replacement"><i class="fa fa-check"></i><b>2.2.1</b> Species replacement</a></li>
<li class="chapter" data-level="2.2.2" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#species-nestedness"><i class="fa fa-check"></i><b>2.2.2</b> Species nestedness</a></li>
<li class="chapter" data-level="2.2.3" data-path="some-fundamentals-by-dr-jen-wood.html"><a href="some-fundamentals-by-dr-jen-wood.html#beta-diversity-analysis-workflow"><i class="fa fa-check"></i><b>2.2.3</b> Beta diversity analysis workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gettingstarted.html"><a href="gettingstarted.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gettingstarted.html"><a href="gettingstarted.html#before-you-can-analyse"><i class="fa fa-check"></i><b>3.1</b> Before you can analyse</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="gettingstarted.html"><a href="gettingstarted.html#get-access-to-computational-resources"><i class="fa fa-check"></i><b>3.1.1</b> Get access to computational resources</a></li>
<li class="chapter" data-level="3.1.2" data-path="gettingstarted.html"><a href="gettingstarted.html#get-file-storage"><i class="fa fa-check"></i><b>3.1.2</b> Get file storage</a></li>
<li class="chapter" data-level="3.1.3" data-path="gettingstarted.html"><a href="gettingstarted.html#learn-some-basic-command-line"><i class="fa fa-check"></i><b>3.1.3</b> Learn some basic command Line</a></li>
<li class="chapter" data-level="3.1.4" data-path="gettingstarted.html"><a href="gettingstarted.html#get-a-data-storage-browser"><i class="fa fa-check"></i><b>3.1.4</b> Get a data storage browser</a></li>
<li class="chapter" data-level="3.1.5" data-path="gettingstarted.html"><a href="gettingstarted.html#get-r-and-r-studio"><i class="fa fa-check"></i><b>3.1.5</b> Get R and R studio</a></li>
<li class="chapter" data-level="3.1.6" data-path="gettingstarted.html"><a href="gettingstarted.html#know-what-environment-managers-are"><i class="fa fa-check"></i><b>3.1.6</b> Know what environment managers are</a></li>
<li class="chapter" data-level="3.1.7" data-path="gettingstarted.html"><a href="gettingstarted.html#explore-the-help-functions"><i class="fa fa-check"></i><b>3.1.7</b> Explore the help functions</a></li>
<li class="chapter" data-level="3.1.8" data-path="gettingstarted.html"><a href="gettingstarted.html#get-a-github-account"><i class="fa fa-check"></i><b>3.1.8</b> Get a GitHub account</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="gettingstarted.html"><a href="gettingstarted.html#a-note-on-rarefying-normalising"><i class="fa fa-check"></i><b>3.2</b> A note on rarefying (normalising)</a></li>
<li class="chapter" data-level="3.3" data-path="gettingstarted.html"><a href="gettingstarted.html#workflows-from-other-lab-groups"><i class="fa fa-check"></i><b>3.3</b> Workflows from other lab groups</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html"><i class="fa fa-check"></i><b>4</b> Miseq library preps</a>
<ul>
<li class="chapter" data-level="4.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#overview"><i class="fa fa-check"></i><b>4.1.1</b> Overview</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#workflow"><i class="fa fa-check"></i><b>4.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-extraction"><i class="fa fa-check"></i><b>4.2.1</b> DNA extraction</a></li>
<li class="chapter" data-level="4.2.2" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-quality-and-concentration"><i class="fa fa-check"></i><b>4.2.2</b> DNA quality and concentration</a></li>
<li class="chapter" data-level="4.2.3" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#dna-normalisation"><i class="fa fa-check"></i><b>4.2.3</b> DNA normalisation</a></li>
<li class="chapter" data-level="4.2.4" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#first-and-second-pcr---amplicon-pcr-and-indexing-pcr"><i class="fa fa-check"></i><b>4.2.4</b> First and second PCR - Amplicon PCR and Indexing PCR</a></li>
<li class="chapter" data-level="4.2.5" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#normalisation-of-indexed-amplicons"><i class="fa fa-check"></i><b>4.2.5</b> Normalisation of indexed amplicons</a></li>
<li class="chapter" data-level="4.2.6" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#pooling"><i class="fa fa-check"></i><b>4.2.6</b> Pooling</a></li>
<li class="chapter" data-level="4.2.7" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#denaturing-of-pool-and-loading-of-the-miseq"><i class="fa fa-check"></i><b>4.2.7</b> Denaturing of pool and loading of the Miseq</a></li>
<li class="chapter" data-level="4.2.8" data-path="miseq-library-preps.html"><a href="miseq-library-preps.html#consumables"><i class="fa fa-check"></i><b>4.2.8</b> Consumables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html"><i class="fa fa-check"></i><b>5</b> From BaseSpace to Qiime2 and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#prerequisites-and-required-files"><i class="fa fa-check"></i><b>5.1.1</b> Prerequisites and required files</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#workflow-1"><i class="fa fa-check"></i><b>5.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#download-fastq-files"><i class="fa fa-check"></i><b>5.2.1</b> Download FASTQ files</a></li>
<li class="chapter" data-level="5.2.2" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#paired-end-manifest-import-step-1"><i class="fa fa-check"></i><b>5.2.2</b> Paired end manifest import (Step 1)</a></li>
<li class="chapter" data-level="5.2.3" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#cutadapt-step-2"><i class="fa fa-check"></i><b>5.2.3</b> Cutadapt (Step 2)</a></li>
<li class="chapter" data-level="5.2.4" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#read-quality-assessment"><i class="fa fa-check"></i><b>5.2.4</b> Read quality assessment</a></li>
<li class="chapter" data-level="5.2.5" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#denoise-paired-end-sequences-with-dada2-step-3"><i class="fa fa-check"></i><b>5.2.5</b> Denoise paired end sequences with dada2 (Step 3)</a></li>
<li class="chapter" data-level="5.2.6" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#taxonomic-classifier-and-assignment-step-4"><i class="fa fa-check"></i><b>5.2.6</b> Taxonomic classifier and assignment (Step 4)</a></li>
<li class="chapter" data-level="5.2.7" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#build-phylogenetic-tree-step-5"><i class="fa fa-check"></i><b>5.2.7</b> Build phylogenetic tree (Step 5)</a></li>
<li class="chapter" data-level="5.2.8" data-path="miseqtoqiime.html"><a href="miseqtoqiime.html#all-steps-combined"><i class="fa fa-check"></i><b>5.2.8</b> All steps combined</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Qiime2R.html"><a href="Qiime2R.html"><i class="fa fa-check"></i><b>6</b> From Qiime2 into R - Initial diagnostics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="Qiime2R.html"><a href="Qiime2R.html#introduction-2"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="Qiime2R.html"><a href="Qiime2R.html#workflow-2"><i class="fa fa-check"></i><b>6.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="Qiime2R.html"><a href="Qiime2R.html#packages"><i class="fa fa-check"></i><b>6.2.1</b> Packages</a></li>
<li class="chapter" data-level="6.2.2" data-path="Qiime2R.html"><a href="Qiime2R.html#import-qiime-files-and-create-a-phyloseq-object"><i class="fa fa-check"></i><b>6.2.2</b> Import qiime-files and create a phyloseq object</a></li>
<li class="chapter" data-level="6.2.3" data-path="Qiime2R.html"><a href="Qiime2R.html#save-the-ps-object-as-an-.rds-file-into-your-working-directory"><i class="fa fa-check"></i><b>6.2.3</b> Save the <code>ps</code> object as an <code>.rds</code> file into your working directory</a></li>
<li class="chapter" data-level="6.2.4" data-path="Qiime2R.html"><a href="Qiime2R.html#initial-filtering"><i class="fa fa-check"></i><b>6.2.4</b> Initial filtering</a></li>
<li class="chapter" data-level="6.2.5" data-path="Qiime2R.html"><a href="Qiime2R.html#proportion-of-asvs-identified"><i class="fa fa-check"></i><b>6.2.5</b> Proportion of ASVs identified</a></li>
<li class="chapter" data-level="6.2.6" data-path="Qiime2R.html"><a href="Qiime2R.html#rarefaction-curve"><i class="fa fa-check"></i><b>6.2.6</b> Rarefaction curve</a></li>
<li class="chapter" data-level="6.2.7" data-path="Qiime2R.html"><a href="Qiime2R.html#prevalence-table"><i class="fa fa-check"></i><b>6.2.7</b> Prevalence table</a></li>
<li class="chapter" data-level="6.2.8" data-path="Qiime2R.html"><a href="Qiime2R.html#export-asv-and-taxa-tables-to-excel"><i class="fa fa-check"></i><b>6.2.8</b> Export ASV and taxa tables to Excel</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plottingdiversity.html"><a href="plottingdiversity.html"><i class="fa fa-check"></i><b>7</b> Plotting diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="plottingdiversity.html"><a href="plottingdiversity.html#introduction-3"><i class="fa fa-check"></i><b>7.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="plottingdiversity.html"><a href="plottingdiversity.html#required-files"><i class="fa fa-check"></i><b>7.1.1</b> Required files</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="plottingdiversity.html"><a href="plottingdiversity.html#workflow-3"><i class="fa fa-check"></i><b>7.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="plottingdiversity.html"><a href="plottingdiversity.html#packages-1"><i class="fa fa-check"></i><b>7.2.1</b> Packages</a></li>
<li class="chapter" data-level="7.2.2" data-path="plottingdiversity.html"><a href="plottingdiversity.html#load-phyloseq-object"><i class="fa fa-check"></i><b>7.2.2</b> Load phyloseq object</a></li>
<li class="chapter" data-level="7.2.3" data-path="plottingdiversity.html"><a href="plottingdiversity.html#pre-filter-rarefy-and-calculate-diversity"><i class="fa fa-check"></i><b>7.2.3</b> Pre-filter, rarefy and calculate diversity</a></li>
<li class="chapter" data-level="7.2.4" data-path="plottingdiversity.html"><a href="plottingdiversity.html#colors"><i class="fa fa-check"></i><b>7.2.4</b> Colors</a></li>
<li class="chapter" data-level="7.2.5" data-path="plottingdiversity.html"><a href="plottingdiversity.html#richness-1"><i class="fa fa-check"></i><b>7.2.5</b> Richness</a></li>
<li class="chapter" data-level="7.2.6" data-path="plottingdiversity.html"><a href="plottingdiversity.html#shannon"><i class="fa fa-check"></i><b>7.2.6</b> Shannon</a></li>
<li class="chapter" data-level="7.2.7" data-path="plottingdiversity.html"><a href="plottingdiversity.html#pielous-evenness"><i class="fa fa-check"></i><b>7.2.7</b> Pielou’s evenness</a></li>
<li class="chapter" data-level="7.2.8" data-path="plottingdiversity.html"><a href="plottingdiversity.html#combined-plotting"><i class="fa fa-check"></i><b>7.2.8</b> Combined plotting</a></li>
<li class="chapter" data-level="7.2.9" data-path="plottingdiversity.html"><a href="plottingdiversity.html#nmds"><i class="fa fa-check"></i><b>7.2.9</b> NMDS</a></li>
<li class="chapter" data-level="7.2.10" data-path="plottingdiversity.html"><a href="plottingdiversity.html#pca"><i class="fa fa-check"></i><b>7.2.10</b> PCA</a></li>
<li class="chapter" data-level="7.2.11" data-path="plottingdiversity.html"><a href="plottingdiversity.html#rda"><i class="fa fa-check"></i><b>7.2.11</b> RDA</a></li>
<li class="chapter" data-level="7.2.12" data-path="plottingdiversity.html"><a href="plottingdiversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>7.2.12</b> Unweighted uniFrac</a></li>
<li class="chapter" data-level="7.2.13" data-path="plottingdiversity.html"><a href="plottingdiversity.html#combined-plotting-1"><i class="fa fa-check"></i><b>7.2.13</b> Combined plotting</a></li>
<li class="chapter" data-level="7.2.14" data-path="plottingdiversity.html"><a href="plottingdiversity.html#save-plots-to-png-and-pdf"><i class="fa fa-check"></i><b>7.2.14</b> Save plots to png and pdf</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="plottingtaxa.html"><a href="plottingtaxa.html"><i class="fa fa-check"></i><b>8</b> Plotting taxonomy</a>
<ul>
<li class="chapter" data-level="8.1" data-path="plottingtaxa.html"><a href="plottingtaxa.html#introduction-4"><i class="fa fa-check"></i><b>8.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="plottingtaxa.html"><a href="plottingtaxa.html#required-files-or-objects"><i class="fa fa-check"></i><b>8.1.1</b> Required files or objects</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="plottingtaxa.html"><a href="plottingtaxa.html#workflow-4"><i class="fa fa-check"></i><b>8.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="plottingtaxa.html"><a href="plottingtaxa.html#packages-2"><i class="fa fa-check"></i><b>8.2.1</b> Packages</a></li>
<li class="chapter" data-level="8.2.2" data-path="plottingtaxa.html"><a href="plottingtaxa.html#load-phyloseq-object-1"><i class="fa fa-check"></i><b>8.2.2</b> Load phyloseq object</a></li>
<li class="chapter" data-level="8.2.3" data-path="plottingtaxa.html"><a href="plottingtaxa.html#check-metadata-contained-in-the-phyloseq-object."><i class="fa fa-check"></i><b>8.2.3</b> Check metadata contained in the phyloseq object.</a></li>
<li class="chapter" data-level="8.2.4" data-path="plottingtaxa.html"><a href="plottingtaxa.html#colors-1"><i class="fa fa-check"></i><b>8.2.4</b> Colors</a></li>
<li class="chapter" data-level="8.2.5" data-path="plottingtaxa.html"><a href="plottingtaxa.html#pre-filtering"><i class="fa fa-check"></i><b>8.2.5</b> Pre-filtering</a></li>
<li class="chapter" data-level="8.2.6" data-path="plottingtaxa.html"><a href="plottingtaxa.html#analysis-and-plotting"><i class="fa fa-check"></i><b>8.2.6</b> Analysis and plotting</a></li>
<li class="chapter" data-level="8.2.7" data-path="plottingtaxa.html"><a href="plottingtaxa.html#prevalance-table"><i class="fa fa-check"></i><b>8.2.7</b> Prevalance table</a></li>
<li class="chapter" data-level="8.2.8" data-path="plottingtaxa.html"><a href="plottingtaxa.html#krona-charts"><i class="fa fa-check"></i><b>8.2.8</b> Krona charts</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="phylofactor.html"><a href="phylofactor.html"><i class="fa fa-check"></i><b>9</b> Phylofactor analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="phylofactor.html"><a href="phylofactor.html#introduction-5"><i class="fa fa-check"></i><b>9.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="phylofactor.html"><a href="phylofactor.html#prerequisites-and-required-files-1"><i class="fa fa-check"></i><b>9.1.1</b> Prerequisites and required files</a></li>
<li class="chapter" data-level="9.1.2" data-path="phylofactor.html"><a href="phylofactor.html#custom-functions"><i class="fa fa-check"></i><b>9.1.2</b> Custom functions</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="phylofactor.html"><a href="phylofactor.html#workflow-5"><i class="fa fa-check"></i><b>9.2</b> Workflow</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="phylofactor.html"><a href="phylofactor.html#load-packages-and-custom-functions"><i class="fa fa-check"></i><b>9.2.1</b> Load packages and custom functions</a></li>
<li class="chapter" data-level="9.2.2" data-path="phylofactor.html"><a href="phylofactor.html#load-phyloseq-object-2"><i class="fa fa-check"></i><b>9.2.2</b> Load phyloseq object</a></li>
<li class="chapter" data-level="9.2.3" data-path="phylofactor.html"><a href="phylofactor.html#pre-filtering-1"><i class="fa fa-check"></i><b>9.2.3</b> Pre-filtering</a></li>
<li class="chapter" data-level="9.2.4" data-path="phylofactor.html"><a href="phylofactor.html#running-phylofactor"><i class="fa fa-check"></i><b>9.2.4</b> Running PhyloFactor</a></li>
<li class="chapter" data-level="9.2.5" data-path="phylofactor.html"><a href="phylofactor.html#reviewing-output"><i class="fa fa-check"></i><b>9.2.5</b> Reviewing output</a></li>
<li class="chapter" data-level="9.2.6" data-path="phylofactor.html"><a href="phylofactor.html#visualising-factors"><i class="fa fa-check"></i><b>9.2.6</b> Visualising factors</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Beginner’s guide to Miseq sequencing and processing of short-read microbial (16S) amplicon sequences</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="some-fundamentals-by-dr-jen-wood" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Some fundamentals by Dr Jen Wood<a href="some-fundamentals-by-dr-jen-wood.html#some-fundamentals-by-dr-jen-wood" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>An introduction to alpha and beta diversity analysis and the pitfalls you need to know</strong></p>
<p>By <a href="https://jenwoodmicro.wordpress.com">Dr Jen WOOD</a> | Microbial community ecologist<br />
<img src="img/JW.jpg" /></p>
<p><strong>Microbial Diversity, how do you describe it?</strong></p>
<ul>
<li><p>How many species (ASVs/ESVs) are there?</p></li>
<li><p>Which species are common and which are rare?</p></li>
<li><p>How many species do two communities share?</p></li>
<li><p>How many species do two communities have different?</p></li>
<li><p>Do two communities share the same common and rare species?</p></li>
</ul>
<p><strong>Some terms and their synonyms</strong></p>
<ul>
<li><p>Community = Microbial composition in each sample</p></li>
<li><p>Species = Operational Taxonomic Unit (OTU) = Amplicon Sequence Variant (ASV) = Exact Sequence Variant (ESV)</p></li>
<li><p>Individual = Read = Sequence read</p></li>
</ul>
<div id="alpha-diversity" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Alpha diversity<a href="some-fundamentals-by-dr-jen-wood.html#alpha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Definition</strong>: Alpha diversity is the diversity of each site (local species pool).</p>
<p>Common metrics such as Shannon-Wiener, Simpson diversity are affected by species <strong>richness</strong> and community <strong>evenness</strong></p>
<div id="richness" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Richness<a href="some-fundamentals-by-dr-jen-wood.html#richness" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Number of species (ASVs/ESVs) in a community.<br />
The more species rich a community is the more diverse that community is.</p>
<ul>
<li><p><strong>Not</strong> affected by the number of INDIVIDUALS (READS/ABUNDANCE).</p></li>
<li><p><strong>Not</strong> affected by taxonomy</p></li>
</ul>
<p><br />
</p>
<p>Two lakes with identical species richness of 5<br />
Just one is more crowded than the other<br />
<img src="img/pond_richness.png" /></p>
<p><br />
</p>
<p>Two communities with an identical OTU richness of 4 but with NO species in common <img src="img/animals_richness.png" /><br />
</p>
</div>
<div id="evenness" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Evenness<a href="some-fundamentals-by-dr-jen-wood.html#evenness" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>How evenly individuals are distributed among the species in a community.<br />
The more even a community is the more diverse that community is.</p>
<ul>
<li><p><strong>IS</strong> affected by the number of individuals (Reads/Abundance).</p></li>
<li><p><strong>Not</strong> affected by taxonomy<br />
</p></li>
</ul>
<p>Examples of how the number of individuals can be distributed among the species in a given community:<br />
</p>
<p><img src="img/animals_evenness.png" /></p>
<p><br />
</p>
</div>
<div id="exercise" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Exercise<a href="some-fundamentals-by-dr-jen-wood.html#exercise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For each pair of communities - which is more diverse and why?</p>
<p><img src="img/petri.png" width="550" /></p>
<p><br />
</p>
<p><strong>Answers</strong><br />
From the top:</p>
<ul>
<li>A is more diverse because species distribution is more even.</li>
<li>A is more diverse because of it greater species richness.</li>
<li>None are more diverse. Both pairs have the same richness and evenness.</li>
</ul>
<p>In general MORE richness = MORE diversity<br />
And MORE evenness = MORE diversity</p>
</div>
<div id="shannon-weiner-diversity" class="section level3 hasAnchor" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> Shannon-Weiner Diversity<a href="some-fundamentals-by-dr-jen-wood.html#shannon-weiner-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Richness and Evenness of a community contribute to its Diversity.<br />
This is a score that can be calculated a number of ways.<br />
The Shannon-Weiner equation below is one of the most commonly used.</p>
<p><img src="img/SW-formula.png" width="350" /></p>
<p><br />
</p>
<p>For this pair of communities - which is more diverse and why?</p>
<p><img src="img/petri2.png" width="550" /></p>
<p><br />
</p>
<p><strong>Example 1</strong><br />
</p>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Species(i)</th>
<th align="center">Number of individuals (p)</th>
<th align="center">pi</th>
<th align="center">Ln(pi)</th>
<th align="center">pi*Ln(pi)</th>
<th align="center">-pi*Ln(pi)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Geobacter spp.</td>
<td align="center">52</td>
<td align="center">0.84</td>
<td align="center">-0.18</td>
<td align="center">-0.15</td>
<td align="center">0.15</td>
</tr>
<tr class="even">
<td align="center">Desulfovibrio spp.</td>
<td align="center">7</td>
<td align="center">0.11</td>
<td align="center">-2.18</td>
<td align="center">-0.25</td>
<td align="center">0.25</td>
</tr>
<tr class="odd">
<td align="center">Clostridium spp.</td>
<td align="center">2</td>
<td align="center">0.03</td>
<td align="center">-3.43</td>
<td align="center">-0.11</td>
<td align="center">0.11</td>
</tr>
<tr class="even">
<td align="center">Sum:</td>
<td align="center">62</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center">0.51</td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p><strong>Example 2</strong><br />
</p>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Species</th>
<th align="center">Number of individuals (p)</th>
<th align="center">pi</th>
<th align="center">Ln(pi)</th>
<th align="center">pi*Ln(pi)</th>
<th align="center">-pi*Ln(pi)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Geobacter spp.</td>
<td align="center">23</td>
<td align="center">0.37</td>
<td align="center">-0.99</td>
<td align="center">-0.37</td>
<td align="center">0.37</td>
</tr>
<tr class="even">
<td align="center">Desulfovibrio spp.</td>
<td align="center">27</td>
<td align="center">0.44</td>
<td align="center">-0.83</td>
<td align="center">-0.36</td>
<td align="center">0.36</td>
</tr>
<tr class="odd">
<td align="center">Clostridium spp.</td>
<td align="center">12</td>
<td align="center">0.19</td>
<td align="center">-1.64</td>
<td align="center">-0.32</td>
<td align="center">0.32</td>
</tr>
<tr class="even">
<td align="center">Sum:</td>
<td align="center">62</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center">1.05</td>
</tr>
</tbody>
</table>
<p><br />
</p>
</div>
</div>
<div id="beta-diversity" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Beta diversity<a href="some-fundamentals-by-dr-jen-wood.html#beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Definition</strong>: The variation in species composition among sites.</p>
<p>Common metrics: (dis)similarity indices such as Bray-Curtis (BC), Unifrac, Aitchison<br />
Affected by <strong>number</strong> of ASVs/ESVs in common, number of ASVs/ESVs not in common and similarity of shared ASV/ESV <strong>abundances</strong> between plots (When data is abundance weighted and not binary transformed)<br />
</p>
<p><br />
E.g. Bray-Curtis dissimilarity:<br />
<img src="img/BC.jpg" width="150" /><br />
</p>
<p>Where:<br />
i &amp; j are the two sites,<br />
S<sub>i</sub> is the total number of specimens counted on site i,<br />
S<sub>j</sub> is the total number of specimens counted on site j,<br />
C<sub>ij</sub> is the sum of only the lesser counts for each species found in both sites.<br />
<br />
</p>
<p>Two communities with identical richness, diversity and biomass can still be a bit dissimilar.<br />
<img src="img/betatank.png" width="550" /><br />
</p>
<p><br />
Two communities with identical richness, diversity and biomass can be highly dissimilar.  <img src="img/betatank2.png" width="550" /><br />
</p>
<p><br />
Changes in beta diversity may not track with changes in alpha diversity. The variation in species composition among sites can arise from <strong>species replacement</strong> (i.e. community turnover) due to the tendency of species to replace each other over an ecological gradient.<br />
It can also come from <strong>species differences</strong> (i.e. community nestedness) due to one community having more species present than another.</p>
<p><br />
</p>
<div id="species-replacement" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Species replacement<a href="some-fundamentals-by-dr-jen-wood.html#species-replacement" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="img/betatank_replace.png" width="550" /><br />
</p>
</div>
<div id="species-nestedness" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Species nestedness<a href="some-fundamentals-by-dr-jen-wood.html#species-nestedness" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="img/betatank_nest.png" width="550" /><br />
</p>
<p> </p>
</div>
<div id="beta-diversity-analysis-workflow" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Beta diversity analysis workflow<a href="some-fundamentals-by-dr-jen-wood.html#beta-diversity-analysis-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="img/betaworkflow.png" /><br />
</p>
<p><strong>Visualisations</strong></p>
<p><img src="img/betaviz.png" /><br />
</p>
<div id="nmnds-and-pcoa" class="section level4 hasAnchor" number="2.2.3.1">
<h4><span class="header-section-number">2.2.3.1</span> NMNDs and PCoA<a href="some-fundamentals-by-dr-jen-wood.html#nmnds-and-pcoa" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Exactly the same data in an NMDS and in a PCoA using Bray-Curtis metric: what’s the difference?</p>
<p><img src="img/beta_nmdspcoa.png" /></p>
<p><br />
</p>
<p><strong>NMDS: non-Metric Multidimensional Scaling</strong></p>
<p><img src="img/beta_ordination.png" width="300" /><br />
</p>
<p><strong>Process</strong></p>
<ol style="list-style-type: decimal">
<li>A half matrix containing the dissimilarity scores between each pair of samples is calculated.<br />
</li>
<li>The dissimilarity relationship between all the samples can be explained in n-dimensional space. Where n = number of samples -1.<br />
</li>
<li>The computer iteratively tries to project this multidimensional relationship space into 2D. If it can do this and preserve the relationships, stress &lt; 0.2. If it can’t do this stress &gt; 0.2 and the ordination is basically random collection of dots.</li>
</ol>
<p><strong>Interpretation</strong><br />
The closer two points are in 2D space the more similar they are. i.e. we looking at grouping.</p>
<p>Axes are arbitrary.</p>
<p>Valid if Kruskal stress values &lt; 0.2.</p>
<p><strong>Pros:</strong> lets the data “speak for itself”. Way the ordination is drawn is completely analogous to the way ANOSIM/PERMANOVA stats are generated.</p>
<p><strong>Cons:</strong> ordinations are not always valid.<br />
</p>
<p><strong>PCoA: Principal Coordinates analysis</strong></p>
<p><img src="img/beta_pcoa.png" width="300" /><br />
</p>
<p><strong>Process</strong></p>
<ol style="list-style-type: decimal">
<li>A half matrix containing the dissimilarity scores between each pair of samples in calculated and ‘placed’ in n-dimensional space.<br />
</li>
<li>An axis is draw through the data in the dimension that has the largest variation.<br />
</li>
<li>Data is projected onto this axis.<br />
</li>
<li>The process is repeated for the direction through the data that has the second greatest amount of variation so the data are now projected onto these two new axis, the PCoA axis.</li>
</ol>
<p><strong>Interpretation</strong><br />
Order of data along axis correspond to largest variation in data set.</p>
<p>Percentage of variation in the ‘data cloud’.</p>
<p><strong>Pros:</strong> Always valid - no stress values; may correlate treatment effect to a tangible level of variation in the data.</p>
<p><strong>Cons:</strong> Axis don’t always explain lots of variation; communities that have ASV/ESV values that introduce large amounts of variation strongly influence the ordination, making patterns of other communities less visible; may need to remove these communities.<br />
</p>
</div>
<div id="binary-transform" class="section level4 hasAnchor" number="2.2.3.2">
<h4><span class="header-section-number">2.2.3.2</span> Binary transform<a href="some-fundamentals-by-dr-jen-wood.html#binary-transform" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Tells us about the influence of rare members and abundance on beta-diversity scores. Binary transformation helps determine why a community is (dis)similar.</p>
<p><img src="img/binarytrans1.png" width="400" /><br />
<img src="img/binarytrans2.png" width="400" /><br />
</p>
<p><strong>Example1:</strong> Communities that share dominate species appear highly similar. What happen when we binary transform?</p>
<p><img src="img/binarytranstank1.png" width="550" /><br />
</p>
<p><img src="img/binarytranstank2.png" width="550" /><br />
</p>
<p><strong>Example2:</strong> The communities with identical richness and biomass are &gt;50% dissimilar due dominant species turnover. What happens when we binary transform?</p>
<p><img src="img/binarytranstank3.png" width="550" /><br />
</p>
<p><img src="img/binarytranstank4.png" width="550" /><br />
</p>
</div>
<div id="testing-beta-diversity-permanova" class="section level4 hasAnchor" number="2.2.3.3">
<h4><span class="header-section-number">2.2.3.3</span> Testing Beta diversity: Permanova<a href="some-fundamentals-by-dr-jen-wood.html#testing-beta-diversity-permanova" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><p>A non-parametric alternative to MANOVA (also called NPMANOVA)</p></li>
<li><p>Generates a pseudo-F stat and F table with p values (like an ANOVA)</p></li>
<li><p>Test hypothesis that there is no difference in group centriods</p></li>
</ul>
<p><img src="img/permanova.png" width="400" /><br />
</p>
<p>PERMANOVA is not done on the output of an ordination but rather on the underlying distance matrices.</p>
<p><img src="img/permaprocess.png" width="400" /><br />
</p>
<p>Beta dispersion describes the spread of a set of points around a centroid.</p>
<p>PERMANOVA can be sensitive to difference in beta-dispersion (so test for this too).</p>
<p><img src="img/dispersion.png" width="500" /><br />
</p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gettingstarted.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/cjvanlissa/gitbook-demo/edit/master/02.1_jenwood.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["gitbook-demo.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
